# Takaoka_Mallat
# Intermittent childhood hyperlipidaemia alters resident arterial macrophages and accelerates atherosclerosis
Minoru Takaoka<sup>1</sup>, Xiaohui Zhao<sup>1\*</sup>, Hwee Ying Lim<sup>2,3\*</sup>, Costan G. Magnussen<sup>4,5,6\*</sup>,
Nadine Suffee<sup>7</sup>, Patricia R. Schrank<sup>8</sup>, Owen Ang<sup>2,3</sup>, Dimitrios Tsiantoulas<sup>1,9</sup>, Felix
Sommer<sup>10</sup>, Sarajo K. Mohanta<sup>11</sup>, James Harrison<sup>1</sup>, Yaxing Meng<sup>6</sup>, Ludivine Laurans<sup>7</sup>,
Feitong Wu<sup>6</sup>, Yuning Lu<sup>1</sup>, Leanne Masters<sup>1</sup>, Stephen A Newland<sup>1</sup>, Laura Denti<sup>12</sup>,
Mingyang Hong<sup>11</sup>, Mouna Chajadine<sup>7</sup>, Markus Juonala<sup>13,14</sup>, Juhani S.
Koskinen<sup>4,5,14,15</sup>, Mika Kähönen<sup>16,17,18</sup>, Tsuneyasu Kaisho<sup>19</sup>, Andreas J.R.
Habenicht<sup>11</sup>, Alain Tedgui<sup>7</sup>, Hafid Ait-Oufella<sup>7</sup>, Tian X Zhao<sup>1</sup>, Meritxell Nus<sup>1</sup>,
Christiana Ruhrberg<sup>12</sup>, Soraya Taleb<sup>7</sup>, Jesse W. Williams<sup>8</sup>, Olli T. Raitakari<sup>4,5,20\*</sup>,
Véronique Angeli<sup>2,3\*</sup>, Ziad Mallat<sup>1,7</sup>
Journal xxx,  [![DOI](https://doi.org/***)](https://doi.org/***) <br>
 
<sup>1</sup> Department of Medicine, Cardiovascular Division, University of Cambridge, Heart and Lung Research Institute, CB2 0BB, Cambridge, UK, <br>
<sup>2</sup> Immunology Translational Research Programme, Yong Loo Lin School of Medicine, Dept. of Department of Microbiology and Immunology, National University of Singapore, Singapore. <br>
<sup>3</sup> Immunology Programme, Life Sciences Institute, National University of Singapore, Singapore.<br>
<sup>4</sup> Research Centre of Applied and Preventive Cardiovascular Medicine; University of Turku, Turku, Finland.<br>
<sup>5</sup> Centre for Population Health Research, University of Turku and Turku University Hospital, Turku, Finland.<br>
<sup>6</sup> Baker Heart and Diabetes Institute, Melbourne, Victoria, Australia.<br>
<sup>7</sup> Université de Paris, Institut National de la Santé et de la Recherche Médicale, U970, PARCC, Paris, France.<br>
<sup>8</sup> Department of Integrative Biology & Physiology, Center for Immunology, University of Minnesota, Minneapolis, USA.<br>
<sup>9</sup> Department of Laboratory Medicine, Medical University of Vienna, Vienna, Austria<br>
<sup>10</sup> Institute of Clinical Molecular Biology, University of Kiel and University Hospital Schleswig Holstein (UKSH), Germany<br>
<sup>11</sup> Institute for Cardiovascular Prevention, Ludwig-Maximilians-33 Universität München(LMU), Munich, Germany.<br>
<sup>12</sup> Institute of Ophthalmology, University College London, UK.<br>
<sup>13</sup> Department of Medicine, University of Turku, Turku, Finland.<br>
<sup>14</sup> Division of Medicine, Turku University Hospital, Turku, Finland.<br>
<sup>15</sup> Department of Medicine, Satakunta Central Hospital, Pori, Finland.<br>
<sup>16</sup> Department of Clinical Physiology, University of Tampere, Tampere, Finland.<br>
<sup>17</sup> Faculty of Medicine and Health Technology, University of Tampere, Tampere,Finland.<br>
<sup>18</sup> Finnish Cardiovascular Research Center Tampere, University of Tampere, Tampere, Finland.<br>
<sup>19</sup> Department of Immunology, Institute of Advanced Medicine, Wakayama Medical University, Japan.<br>
<sup>20</sup> Department of Clinical Physiology and Nuclear Medicine, Turku University Hospital, Turku, Finland.<br>

\* Contributed equally <br>
**Corresponding authors:** Ziad Mallat (zm245@medschl.cam.ac.uk)<br>

Code Release for this github: [![DOI](https://zenodo.org/badge/***)](https://zenodo.org/badge/latestdoi/***)

## Abstract
Hyperlipidaemia constitutes a major risk factor of atherosclerotic cardiovascular disease (ASCVD). Recent observations indicate that the risk of incident cardiovascular events depends on cumulative lifetime exposure to low-density lipoprotein cholesterol (LDL-C) and, independently, on the time course of exposure to LDL-C, with early exposure being associated with a higher cardiovascular risk. Furthermore, periodic variations in circulating LDL-C are strongly associated with ASCVD outcomes, independently of mean LDL-C levels. However, the precise mechanisms responsible for this increased ASCVD risk are not understood. Here, we make the unexpected observation that early intermittent feeding of mice with a high-cholesterol Western-type diet (WD) substantially accelerates atherosclerosis compared with late exposure to WD, despite similar cumulative plasma LDL-C levels. We find that early intermittent hyperlipidaemia substantially alters the number and homeostatic phenotype of resident-like arterial macrophages. Macrophage genes with altered expression in response to early intermittent hyperlipidaemia are enriched for genes linked to human ASCVD in genome-wide association studies. We show that resident arterial macrophages are atheroprotective and identify new biological pathways whose alteration in arterial macrophages leads to acceleration of atherosclerosis. Finally, data analyses from the Young Finns Study show that exposure to cholesterol in childhood is significantly associated with both the incidence and size of carotid atherosclerotic plaques in mid-adulthood, independently of cumulative lifetime exposure. In summary, our results identify exposure to cholesterol early in life as a strong determinant of accelerated atherosclerosis and provide insight into the underlying biological mechanisms. The current paradigm is that atherosclerosis is an age-associated disease. Our results provide strong evidence that atherosclerosis in adulthood is determined in childhood and highlight the importance of optimal control of hyperlipidaemia early in life. This knowledge will be essential to the design of novel and effective therapeutic strategies to combat atherosclerotic cardiovascular disease.
## Data Availability

* RNASeq data for cWD and iWD: EMBL-EBI ArrayExpress <br>
  **E-MTAB-12759**(https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-12759/) <br>
Illumina HiSeq 4000 platform, single-end reads (SE50), 2 Runs
* RNASeq data for cWD Nrp1 KO: EMBL-EBI ArrayExpress <br>
  **E-MTAB-12761**(https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-12761/)<br>
Illumina NovaSeq6000 platform, paired-end reads (PE50)
* single Cell RNASeq data: public data, please check the supplementary Tables (STable 17, STable 18).

## Corresponding Data/Tables are given below
Supplementary_Tables_iWD paper.xlsx[[xlsx](./Figures_Tables/Supplementary Tables_iWD paper.xlsx)]


## RNASeq analysis Methods

RNA sequencing analyses. iWD and cWD for Ldlr−/− mice single end, unstranded sequencing was performed using SMART-Seq v4 plus nextera (Ultra Low Input RNA Kit (Takara)) with read lengths of 50bp. Raw fastq data have been deposited in ArrayExpress with accession number E-MTAB-12759 (https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-12759). There are two separate experiments for this study with iWD and cWD groups, each group has three independent pools per experiment, and each pool comprising 3 different mice. All raw data were merged and analysed together, with a total of 6 different pools for each group. Sample summary table is in Supplementary Table 13a. The QC, alignment and gene quantification analysis were run using nextflow (version 21.05.0.edge)<sup>38</sup> rnaseq pipeline with nf-core/rnaseq (version 3.2, https://nf-co.re/rnaseq)<sup>39</sup> with the option “--aligner star_salmon” and Ensembl reference genome with the annotation for mouse GRCm39. The alignment results are with a mean of 72% reads uniquely mapping and mean of 34M reads per sample (see Supplementary Table 13). Corresponding software/packages and versions are listed in Supplementary Table 14. All scripts, with details of software versions, expression raw count files and results are freely available from https://github.com/Takaoka_Mallat (DOI:xxxx).<br>
Counts extracted with htseq-counts were used to perform the differential genes (DEs) analysis using DESeq2 (version 1.36.0; R package, version 4.2.1)(https://www.r-project.org/ and <sup>40</sup>). There are 746 genes identified as significantly differential expressed genes (DEGs) when comparing iWD to cWD group using the cut-off threshold padj < 0.05 and |log2FoldChange| >= 1(which means absolute 2-fold changes). The list of significant DEGs is in Supplementary Table 1 and a volcano plot is presented in Fig. 2b. Gene Ontology (GO) and KEGG pathway analysis were performed using R package clusterProfiler (version 4.4.4)<sup>41,42</sup> on the significant DEGs (N=746), all identified pathways are listed in Supplementary Table 2, with 10 selected pathways plotted in Fig. 2a.
Transcription Factor (TF) binding motif enrichment analysis was performed using Rcistarget package (version 1.16.0). Motif rankings for  500 bp upstream of TSS and 100bp downstream was
used in the analysis (mouse motif collection version 9, ‘mc9nr’, with
24453 motifs). Both significant DEGs and redefined (padj>0.1 & |log2foldChange |< 1) non-significant DEGs TFs analysis were performed. We identified 73 common enriched binding motifs corresponding to 27 TFs (see Supplementary Table 4) and we performed two samples proportion z-test and adjusted the pvalues using the Benjamini & Hochberg method. We generated a boxplot by plotting the difference of NES (normalised enrichment score) between significant and non-significant DEGs. Additionally, “SPIC” was selected and GO analysis performed on significant DEGs (N=199) showing enrichment for SPIC binding motifs (see Fig. 2g and Supplementary Table 5).<br>
To assign significant DEGs to specific mouse macrophage cell types, we performed integrated analysis of the published single-cell RNA-Seq data on murine atherosclerosis. We selected three main mouse studies from the paper by Zernecke et al.19 (see Supplementary Table 15, and Extended data Fig. 10). Both individual and integrated study analyses were applied for the data using R package Seurat (version 4.2.0)<sup>43</sup>. We applied FeaturePlot for a set of key mouse macrophage genes (Csf1r, Sepp1, Pf4, F13a1, Folr2, Nrp1, Lyve1, Cd163, Ednrb, Timd4, Phactr1, Egr1,Lgals3, Ctsd, Cxcl2, Ccl4,Trem2,  Spp1, Cd9,  Gpnmb,  Il1b, Cd74, Stat1, Ifit3, Mnda, Irf7, Isg15,  Ifit2 , Fcgr4) to identify the macrophage cluster (Extended Data Fig 10a). Then, we re-clustered the macrophage cell types into five main subtypes of macrophages identified as Adventitia/Resident-like, IFN-, Inflammation-, MacAIR-, and TREM2-like (Extended Data Fig 10b). MacAIR- and TREM2-like macrophages shared common markers and were grouped together for some analyses.<br>
GWAS analysis was performed using the extracted list related to atherosclerosis/CAD (Coronary Artery Disease) from GWAS catalog (https://www.ebi.ac.uk/gwas/) (Supplementary Table 16) and from supplementary table 13 of Tcheandjieu et al.<sup> 24 </sup>. The GWAS list human symbols were converted to mouse symbols using g:Profiler2 orthology (version 0.2.1)<sup>44</sup> in R (Supplementary Table 17). Then we performed the overlap analysis between the GWAS list and our DEGs list (significant DEGs and non-significant DEGs, see Fig. 4a-4b and Supplementary Tables 6 and 7). To assign the 210 significant DEGs that are linked to coronary artery GWAS to specific macrophage cell types, we performed integrated analysis of the published single-cell RNA-Seq data on human atherosclerosis. Six human studies from cardiovascular PlaqView were selected (https://www.plaqview.com/, Supplementary Table 18). Integrated analysis was performed first using R package Seurat (version 4.2.0)<sup>43</sup>, and the macrophage cluster was identified by applying the FeaturePlot for a set of key human macrophage genes (CSF1R, PF4, FOLR2, NRP1, LYVE1, LGALS3, CXCL2, SPP1, GPNMB,  IL1B, APOC1, APOE, FCGR4, IFIT2, FABP4) (Extended Data Fig. 11). Then, we re-clustered the macrophage cell types into five main subtypes of macrophages identified as Adventitia/Resident-like, IFN-, Inflammation-, Foamy-like, and Mixed subtypes (Extended Data Fig 11). Then we assigned our GWAS-related 210 significant DEGs (Supplementary Table 6) to each macrophage cell type key markers list (see Fig. 4b and Supplementary Table 7). <br>
For both human and mouse macrophages, we identified NRP1high and NRP1low groups, then performed gene ontology pathway analysis using enrichR (R package, version 3.1)<sup>45-47</sup> (Supplementary Tables 9 and 10).<br>
Paired-end (PE50), reverse stranded sequencing for Nrp1 WT versus Nrp1 KO peritoneal macrophages in reconstituted Ldlr−/− male mice was performed using Illumina NovaSeq6000 platform. Raw fastq data have been deposited in ArrayExpress with accession number E-MTAB-12761 (http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-12761). DEGs analysis is the same workflow as mentioned for iWD vs cWD. Sample summary and mapping statistics table is in Supplementary Table 13b. There were 113 significant DEGs identified (Supplementary Table 12). Gene Ontology was performed using clusterProfiler (R package, version 4.4.4) with the 113 DEGs, and the list of pathways is given in Supplementary Table 11.<br>


## RNASeq Analysis <br>
### (QC->DEGs->GeneOntology + Transcriptome Factor analysis)

## Step 1: QC and Alignment pipeline
(nextflow version 21.05.0, nf-core/rnaseq version 3.2)

#### Sample sheet input for nextflow pipeline
Macrophage_FirstBatch_nextflow_SampleTable.csv[[csv](Macrophage_FirstBatch_nextflow_SampleTable.csv)] <br>
Macrophage_SecondBatch_nextflow_SampleTable.csv[[csv](Macrophage_SecondBatch_nextflow_SampleTable.csv)] <br>
NRP1-nextflow_SampleTable.csv[[csv](NRP1-nextflow_SampleTable.csv)] <br>

#### Nextflow run pipeline command in HPC
```shell
nextflow run nf-core/rnaseq -bg -profile singularity -r 3.2 \
                            --skipBiotypeQC --input CAD_mt709_0001-Macrophage-nextflow_SampleTable.csv \
                            --aligner star_salmon --fasta GRCm39.fa.gz \
                            --gtf GRCm39.gtf.gz --gtf_extra_attributes 'gene_name' \
                            --outdir  Macrophage_Results --multiqc_title CAD_mt709_0001-Macrophage_NF_Ensembl_GRCm39 \
                            --email xz289@cam.ac.uk -with-report Macrophage_Results/report.html &> Macrophage_Results/nextflow_command.log &
nextflow run nf-core/rnaseq -bg -profile singularity -r 3.2 \
                            --skipBiotypeQC --input CAD_mt709_0001-NRP1-nextflow_SampleTable.csv \
                            --aligner star_salmon --fasta GRCm39.fa.gz \
                            --gtf GRCm39.gtf.gz --gtf_extra_attributes 'gene_name' \
                            --outdir  NRP1_Results --multiqc_title CAD_mt709_0001-NRP1_NF_Ensembl_GRCm39 \
                            --email xz289@cam.ac.uk -with-report NRP1_Results/report.html &> NRP1_Results/nextflow_command.log &
```

## Step 2: Differential Analysis
Using R (v4.2.1) with DESeq2 (v1.36.0) & Gene Ontology Analysis

The analysis code is <br>
#### Macrophage_DESeq_GO_Analysis.R[[Rscript](./Scripts/Macrophage_DESeq_GO_Analysis.R)]
#### Nrp1_DESeq_GO_Analysis.R[Rscript](./Scripts/Nrp1_DESeq_GO_Analysis.R)]

                 Design formula ~ condition
* significant cut-off threshold is padj < 0.05 & abs(log2FoldChange) >= 1
* DESeq2 analysis for iWD vs cWD (merge two batches, significant DEGs STable 1)
* DESeq2 analysis (Nrp1 KO vs WT for cWD, significant DEGs STable 12)

## Step 3: Gene Ontology Analysis
* Input data are the significant DEGs from DESeq analysis.
* Using R package clusterProfiler (version 4.4.4), based on Biological Process mainly.
* GeneOntology results are given in Supplementary Table STable 2 and STable 11 and Figure2a.
* GeneOntology results for NRP1 KOvsWT are illustrated in both Fig4j and Supplementary Table S11.

## Step 4: Transcription Factor (TF) binding motif enrichment analysis
* Significant DEGs and non-significant (p > 0.1 & abs(log2FoldChange) < 1)
* R package Rcistarget (version 1.16.0) with motif rankings for  500 bp upstream of TSS and 100bp downstream was used in the analysis (mouse motif collection version 9, ‘mc9nr’, with
24453 motifs).
* Common motif listed in Supplementary Table STable 4.
* Spic was selected and perform the GeneOntology analysis for its relating sig DEGs. see Supplementary Table S5.

## Step 5: GWAS Analysis
* GWAS list relating to CVD (atherosclerosis) from EBI GWAS, see Supplementary Table STable 7 and also joint the table from supplementary table 13 of Tcheandjieu et al.<sup> 24 </sup>.
* Overlap with iWDvscWD sig/refined non-sig DEGs, barplot presented in Figure 4a and Supplementary Table S6.
* GWAS data is also overlap with single cell integrating analysis of mouse for macrophage subtypes, see Supplementary Table S7.


## Intergrated public single cell RNASeq Analysis
3 Mouse and 6 human public single cell RNASeq data were applied for identified macrophage clusters and then reclustering for detecting specific macrophage subtypes. Data summary is listed in Supplementary Table S17,18. <br>
R package Seurat (version 4.2.0) was used to the clustering analysis. R scripts
#### Mouse_integrated_scRNASeq_Analysis.R[[Rscript](./Scripts/Mouse_integrated_scRNASeq_Analysis.R)]
#### Human_integrated_scRNASeq_Analysis.R[[Rscript](./Scripts/Mouse_integrated_scRNASeq_Analysis.R)]
And each individual study analysis R scripts are given as below.
#### Mouse_integrated_scRNASeq_Analysis.R[[Rscript](./Scripts/Mouse_integrated_scRNASeq_Analysis.R)]
#### Human_integrated_scRNASeq_Analysis.R[[Rscript](./Scripts/Mouse_integrated_scRNASeq_Analysis.R)]


### Corresponding Figures are given below

| Figure | Link   | Images |Legend |
| ----------------------------- | --- |----|----|
|Fig2a | [[PDF](Figures_Tables/Fig2a-MacMer_GOBP_sel_Barplot_Up_DW_Nov_2022.pdf)] |<IMG SRC="Figures_Tables/Fig2a-MacMer_GOBP_sel_Barplot_Up_DW_Nov_2022.pdf.png" width=800px>| Selected Biological Process pathways bar plot for iWD vs cWD.|
|Fig2b | [[PDF](Figures_Tables/Fig2b-MacMer_VolcanoPlot_selmarkers_05_Jan_2023.pdf)] |<IMG SRC="Figures_Tables/Fig2b-MacMer_VolcanoPlot_selmarkers_05_Jan_2023.pdf.png" width=800px>|  Volcano plot for DEGs iWD vs cWD, highlight the MacAir and Reslike Macrophage markers.|
|Fig2f | [[PDF](Figures_Tables/Fig2f-TFs_sig_Nsig_NES_difference_boxplot_Dec_2022.pdf)] |<IMG SRC="Figures_Tables/Fig2f-TFs_sig_Nsig_NES_difference_boxplot_Dec_2022.pdf.png" width=800px>| Transcription factor analysis for DEGs and nonDEGs Normalised enrichment score (NES) difference boxplot.|
|Fig2g | [[PDF](Figures_Tables/Fig2g-Spic_selectedPathway_Barplot_Dec_2022.pdf)] |<IMG SRC="Figures_Tables/Fig2g-Spic_selectedPathway_Barplot_Dec_2022.pdf.png" width=800px>| TF "Spic" relating DEGs selected Biological process enriched pathways bar plot.|
|Fig4a | [[PDF](Figures_Tables/Fig4a-GWAS_sig_Nsig_p01_l2fc1_BarPlot_Jan_2023.pdf)] |<IMG SRC="Figures_Tables/Fig4a-GWAS_sig_Nsig_p01_l2fc1_BarPlot_Jan_2023.pdf.png" width=800px>| Barplot for DEGs and nonDEGS overlap proportion with CAD relating GWAS data.|
|Fig4b | [[PDF](Figures_Tables/Fig4b-MacMer_sigDEGs_overlapCelltype_gwas_piechart_group3_Jan_2022.pdf)] |<IMG SRC="Figures_Tables/Fig4b-MacMer_sigDEGs_overlapCelltype_gwas_piechart_group3_Jan_2022.pdf.png" width=800px>|  pie chart to show the proportion of the DEGs overlap with GWAS with the macrophage sub-types.|
|Fig4c | [[PDF](Figures_Tables/Fig4c-Fernandez_2020_Human_SymAsym_VolcanoPlot_Jan_2023_gwas.pdf)] |<IMG SRC="Figures_Tables/Fig4c-Fernandez_2020_Human_SymAsym_VolcanoPlot_Jan_2023_gwas.pdf.png" width=800px>|  Volcano plot using Fernandez (2020) human data for SymvsAsym plaques data. |
|Fig4d | [[PDF](Figures_Tables/Fig4d-Mouse_integrated_selMac_dotplot_selMarkers_09_02_2023_N1.pdf)] |<IMG SRC="Figures_Tables/Fig4d-Mouse_integrated_selMac_dotplot_selMarkers_09_02_2023_N1.pdf.png" width=800px>|  Dotplot for selected markers of Mouse public atherosclerosis integrated scRNASeq data.|
|Fig4e | [[PDF](Figures_Tables/Fig4e-Human_integrated_selMac_dotplot_selMarkers_01_02_2023_N1.pdf)] |<IMG SRC="Figures_Tables/Fig4e-Human_integrated_selMac_dotplot_selMarkers_01_02_2023_N1.pdf.png" width=800px>| Dotplot for selected markers of Human public atherosclerosis integrated scRNASeq data.|
|Fig4f | [[PDF](Figures_Tables/Fig4f-Human_scRNASeq_Selcted_NRP1_highvslow_Barplot_N.pdf)] |<IMG SRC="Figures_Tables/Fig4f-Human_scRNASeq_Selcted_NRP1_highvslow_Barplot_N.pdf.png" width=800px>|  Selected Biological process Gene ontology pathways for human NRP1 high clusters vs NRP1 low clusters. |
|Fig4j| [[PDF](Figures_Tables/Fig4j-Mouse_cWD_KOvsWT_Selcted_NRP1_highvslow_Barplot_N.pdf)] |<IMG SRC="Figures_Tables/Fig4j-Mouse_cWD_KOvsWT_Selcted_NRP1_highvslow_Barplot_N.pdf.png" width=800px>|  Selected Biological process Gene ontology pathways for mouse cWD NRP1 KOvsWT. |






## Software R Versions & Methods

````
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Ventura 13.2.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     Mersenne-Twister
 Normal:  Inversion
 Sample:  Rounding

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] ComplexHeatmap_2.5.1        apeglm_1.8.0
 [3] limma_3.42.2                ggalt_0.4.0
 [5] dplyr_0.8.5                 plyr_1.8.6
 [7] biomaRt_2.42.1              reshape2_1.4.4
 [9] ggrepel_0.8.2               pheatmap_1.0.12
[11] cowplot_1.0.0               RColorBrewer_1.1-2
[13] ggplot2_3.3.0               DESeq2_1.26.0
[15] SummarizedExperiment_1.16.1 DelayedArray_0.12.3
[17] BiocParallel_1.20.1         matrixStats_0.56.0
[19] Biobase_2.46.0              GenomicRanges_1.38.0
[21] GenomeInfoDb_1.22.1         IRanges_2.20.2
[23] S4Vectors_0.24.4            BiocGenerics_0.32.0
[25] clusterProfiler_3.14.3      rrvgo_1.2.0
.....
````

## Contact

Contact Xiaohui Zhao (xz289 -at- cam.ac.uk) 
