---
title: "Athero_Mouse_scRNA_public_Integrating_12_06_2023"
author: "Xiaohui Zhao"
date: "2023-06-12, revised"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE,error = FALSE,warning = FALSE,cache = FALSE)
```

### ** Load the datasets** and Basic settings


```{r}
setwd('/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/')
list.files()
library(Seurat)
library(ggplot2)
library(viridis)
library(RColorBrewer)
library(dplyr)
library(cowplot)
library(tidyr)
library(tidyverse)
# library(xlsx)
library(patchwork)

```


```{r}
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Cochain_Chow.Robj")
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Cochain_11weeks.Robj")
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Cochain_20weeks.Robj")
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Kim_Ldlr.Robj") 
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Cochain_A11A12.Robj")
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Williams_21days_HFD.Robj")
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Williams_C57.Robj")
```


```{r}

cellids <- c("Cochain_Ldlr_11_weeks_HFD", "Cochain_Ldlr_20_weeks_HFD", "Cochain_A11A12", "Cochain_Chow",  "Kim_Ldlr_10_weeks_HFD", "Williams_21days_HFD", "Williams_C57")

merge <- merge(x=Cochain_11weeks, y=list(Cochain_20weeks, Cochain_A11A12, Cochain_Chow,  Kim_Ldlr, Williams_21days_HFD, Williams_C57), add.cell.ids=cellids)
 
AtheroUpdate.list <- SplitObject(merge, split.by = "Protocol")
AtheroUpdate.list <- AtheroUpdate.list[cellids]
head(AtheroUpdate.list)

```

```{r}
for (i in 1:length(AtheroUpdate.list)) {
    AtheroUpdate.list[[i]] <- NormalizeData(AtheroUpdate.list[[i]], verbose = FALSE)
    AtheroUpdate.list[[i]] <- FindVariableFeatures(AtheroUpdate.list[[i]], selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
}
```

```{r}
reference.list <- AtheroUpdate.list
features <- SelectIntegrationFeatures(object.list = reference.list)
AtheroUpdate.anchors <- FindIntegrationAnchors(object.list = reference.list, dims = 1:30, anchor.features = features)
```

```{r}
save(AtheroUpdate.anchors, file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.anchors.Robj")
```




```{r}
load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.anchors.Robj")
```


```{r}
AtheroUpdate.integrated <- IntegrateData(anchorset = AtheroUpdate.anchors, dims = 1:30)
```


```{r}
DefaultAssay(AtheroUpdate.integrated) <- "integrated"
```


```{r}
save(AtheroUpdate.integrated, file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.Robj")

```

Reviewed after comments back, using python scVI reintegrated and select the subset of macrophage, 
```{r}
load("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.Robj")
mygenes <- rownames(GetAssay(AtheroUpdate.integrated))
AtheroUpdate.integrated <- ScaleData(AtheroUpdate.integrated, features = mygenes, verbose = FALSE)
AtheroUpdate.integrated <- RunPCA(AtheroUpdate.integrated, npcs = 30, verbose = FALSE)
```


```{r}
AtheroUpdate.integrated <- RunUMAP(AtheroUpdate.integrated, dims = 1:30)
```
Find clusters and using "clustree" to decide the clusters with different resolutions.
```{r, fig.width = 6, fig.height = 8}
#AtheroUpdate.integrated  <- FindNeighbors(AtheroUpdate.integrated, dims = 1:30)
AtheroUpdate.integrated  <- FindNeighbors(AtheroUpdate.integrated, dims = 1:30, reduction = "integrated.scvi")
AtheroUpdate.integrated  <- FindClusters(AtheroUpdate.integrated, resolution = 0)
for(res in c(0.3,0.4,0.5,0.6,0.7,0.8,0.9,1))
{
  message(paste0("+--- Method 0 : Finding Clusters at ", res, " resolution ---+"))
  AtheroUpdate.integrated <- FindClusters(AtheroUpdate.integrated, resolution = res, print.output = FALSE) 
}

library(clustree)
mat.convert  <- AtheroUpdate.integrated@meta.data
mat.convert.numeric <- apply(mat.convert[,grep("integrated_snn_res.", colnames(mat.convert))], 2, as.numeric)
clust.input <- as.data.frame(mat.convert.numeric)
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Clustree_genecut200_cellcut3_res0_1.pdf", width = 15, height = 20)
clustree(clust.input, prefix ="integrated_snn_res.")
dev.off()

```


```{r}
AtheroUpdate.integrated <- FindClusters(AtheroUpdate.integrated, resolution = 0.3)
UMAPPlot(AtheroUpdate.integrated, label=T)

```

```{r }
#AtheroUpdate.integrated[["CCA_Clusters_03"]] <- Idents(object = AtheroUpdate.integrated)
AtheroUpdate.integrated[["scvi_Clusters_03"]] <- Idents(object = AtheroUpdate.integrated)
```


```{r, fig.width = 16, fig.height = 6}
UMAPPlot(AtheroUpdate.integrated,  split.by="Protocol", ncol=4)+NoLegend()
```



### ** Find Markers at 0.3 resolution, cell cycle and annotate populations**

```{r}
AtheroUpdate.integrated.markers<- FindAllMarkers(AtheroUpdate.integrated, min.pct = 0.3, logfc.threshold = 0.2, max.cells.per.ident = 250, only.pos = T)
```

```{r}
write.csv(AtheroUpdate.integrated.markers,file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.markers.res03.csv")

save(AtheroUpdate.integrated.markers,file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.markers.res03.Robj")

load(file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.markers.res03.Robj")
```

```{r}
#AtheroUpdate_CCA_Average <- AverageExpression(AtheroUpdate.integrated, return.seurat = T)
AtheroUpdate_scvi_Average <- AverageExpression(AtheroUpdate.integrated, return.seurat = T)
```

```{r}
#save(AtheroUpdate_CCA_Average,file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate_CCA_Average_res03.Robj")
#save(AtheroUpdate_scvi_Average,file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate_scvi_Average_res03.Robj")

```

```{r, fig.width = 6, fig.height = 10}
Markers <- AtheroUpdate.integrated.markers %>% filter(pct.1 > 0.3)

top20= AtheroUpdate.integrated.markers%>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)

top5 <- Markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)
#heat<- DoHeatmap(AtheroUpdate_CCA_Average, features = top5$gene, group.bar = T, size = 4, angle = 90,draw.lines=F, raster = F) + NoLegend()
heat<- DoHeatmap(AtheroUpdate_scvi_Average, features = top5$gene, group.bar = T, size = 4, angle = 90,draw.lines=F, raster = F) + NoLegend()
heat
```

** Load cell cycle genes and convert to mouse nomenclature**

```{r}
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

s.genes <-tolower(s.genes)
g2m.genes <- tolower(g2m.genes)

simpleCap <- function(x) {
  s <- strsplit(x, " ")[[1]]
  paste(toupper(substring(s, 1,1)), substring(s, 2),
      sep="", collapse=" ")
}

s.genes <- sapply(s.genes, simpleCap)
g2m.genes <-sapply(g2m.genes, simpleCap)

```


```{r}
AtheroUpdate.integrated <- CellCycleScoring(AtheroUpdate.integrated, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
```

```{r, fig.width = 5, fig.height = 5}
UMAPPlot(AtheroUpdate.integrated, label=T)
```
UMAP for cell cycle score.
```{r,  fig.width = 10, fig.height = 4}

g2mphase <- FeaturePlot(AtheroUpdate.integrated, features=c("G2M.Score"))+scale_color_viridis(option="inferno")

sphase <-FeaturePlot(AtheroUpdate.integrated, features=c("S.Score"))+scale_color_viridis(option="inferno")

plot_grid(sphase, g2mphase)
```

**Check dissociation induced immediate early genes**

```{r}
ieg.genes <-list( c("Jun", "Junb", "Jund", "Fos", "Fosb", "Atf3", "Hsp90aa1", "Hsp90ab1","Egr1", "Hspa1a", "Hspa1b", "Hspa8", "Zfp36", "Cebpb", "Cebpd", "Socs3", "Hspe1", "Dusp1"))
```
Add the IEG score to the metadata.

```{r}
AtheroUpdate.integrated <- AddModuleScore(AtheroUpdate.integrated, features = ieg.genes, name="IEG_Score", ctrl = 10)
```

```{r}
head(AtheroUpdate.integrated@meta.data)
```


```{r, fig.width = 5, fig.height = 4}

FeaturePlot(AtheroUpdate.integrated, features=c("IEG_Score1"), min.cutoff = "q5", max.cutoff = "q95")+scale_color_viridis(option="inferno")+labs(title="Immediate Early Gene expression score")
```


** Annotate at 0.3 resolution**

```{r}
head(AtheroUpdate.integrated@meta.data)
#Idents(AtheroUpdate.integrated)=AtheroUpdate.integrated[["CCA_Clusters_03"]]
Idents(AtheroUpdate.integrated)=AtheroUpdate.integrated[["scvi_Clusters_03"]]
levels(AtheroUpdate.integrated)
save(AtheroUpdate.integrated,file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.res03.Robj")

```
Generate UMAPs for publication Figs&SFigs.

```{r, fig.width= 5, fig.height= 4}
Mac_genes <- c("Sepp1","Pf4", "F13a1", "Folr2", "Nrp1", "Lyve1","Cd163","Ednrb", "Timd4",
               "Phactr1", "Egr1","Lgals3", "Ctsd", "Cxcl2", "Ccl4","Trem2",  "Spp1", "Cd9",
               "Gpnmb",   "Adgre1", "Il1b", "Cd74", "Stat1", "Ifit3","Mnda", "Irf7","Isg15",
               "Ifit2" , "Fcgr4", "Fcgr1")

message("+------------------Extended Fig10a---------------------------------------------+")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ExtFig10a_Mouse_integrated_all_UMAP_MacGenes_Add_Adgre1_Fcgr1_Jun_2023.pdf", width = 9, height = 4, onefile = T)
umap_inte <- UMAPPlot(AtheroUpdate.integrated, label=T) + theme(legend.position = "none")
mac_int   <- list(Mac_genes)
object    <- AtheroUpdate.integrated
DefaultAssay(object) <- "RNA"
object    <- AddModuleScore(object = object, features = mac_int, name = "Mac_Genes")
macF      <- FeaturePlot(object = object, features = "Mac_Genes1") +
     scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
plot_grid(umap_inte, macF, ncol = 2)
dev.off()
```
Key macrophages featurePlot to identify the Macrophage clusters.

```{r, fig.width= 10, fig.height= 8}
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Mouse_integrated_FeaturePlot_Adgre1_Fcgr1_Csf1r_Pf4_Jun_2023.pdf", onefile = T)

macAdgre1 <- FeaturePlot(object = AtheroUpdate.integrated, features = "Adgre1") +
     scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
macFcgr1 <- FeaturePlot(object = AtheroUpdate.integrated, features = "Fcgr1") +
     scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
macCsf1r <- FeaturePlot(object = AtheroUpdate.integrated, features = "Csf1r") +
     scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
macPf4 <- FeaturePlot(object = AtheroUpdate.integrated, features = "Pf4") +
     scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
plot_grid(macAdgre1, macFcgr1, macCsf1r, macPf4, ncol = 2)
dev.off()
```
Try to identify the macrophage subset groups from the 16 clusters using the markers in STable1, and recluster the subset. 

```{r}
AtheroUpdate_MAC <- subset(AtheroUpdate.integrated, idents=c(0,1,2,7,3,6,14))
save(AtheroUpdate_MAC, file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/AtheroUpdate.integrated.res03.addc7.Macrophage.Robj")
```
Recluster the macrophage subset.

```{r, fig.width = 10, fig.height = 4}
MAC_Reclustered <- AtheroUpdate_MAC
DefaultAssay(MAC_Reclustered) <- "integrated"
mygenes <- rownames(GetAssay(MAC_Reclustered))

MAC_Reclustered <- ScaleData(MAC_Reclustered, features= mygenes, verbose = FALSE)
MAC_Reclustered <- RunPCA(MAC_Reclustered, npcs = 30, verbose = FALSE)
MAC_Reclustered <- RunUMAP(MAC_Reclustered, reduction = "pca", dims = 1:30)
MAC_Reclustered <- FindNeighbors(MAC_Reclustered, dims = 1:30)
MAC_Reclustered <- FindClusters(MAC_Reclustered, resolution = 1.2)

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Macrophage_c01236714_Ldlr_ChowHFD_res12_c19_UMAP.pdf", width=10, height=4)
umap_all <- UMAPPlot(MAC_Reclustered, label=T)
macair_highlight <- rownames(MAC_Reclustered@meta.data[MAC_Reclustered@meta.data$Williams_C57_ID=="Mac_AIR",])
umap_macair<- UMAPPlot(MAC_Reclustered, label=F, cells.highlight=list(macair_highlight))
plot_grid(umap_all, umap_macair, ncol = 2)
dev.off()

MAC.integrated.markers <- FindAllMarkers(MAC_Reclustered, min.pct = 0.3, logfc.threshold = 0.20, only.pos = T)

MAC.top20 = MAC.integrated.markers%>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
MAC_Reclustered@meta.data$Treatment <- ifelse(MAC_Reclustered@meta.data$Protocol == "Cochain_Chow"|MAC_Reclustered@meta.data$Protocol == "Williams_C57", "Ctrl", "HFD")

MAC.conserve.markers9 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 9,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers7 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 7,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers13 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 13,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers10 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 10,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers1 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 1,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers12 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 12,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers2 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 2,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers6 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 6,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers0 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 0,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
MAC.conserve.markers3 <- FindConservedMarkers(MAC_Reclustered,
                     ident.1 = 3,
                     grouping.var = "Treatment",
                     only.pos = TRUE)
#MAC_Reclustered_CCA_Average <- AverageExpression(MAC_Reclustered, return.seurat = T)
MAC_Reclustered_scvi_Average <- AverageExpression(MAC_Reclustered, return.seurat = T)
MAC.top5 = MAC.integrated.markers%>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)
#heatmac <- DoHeatmap(MAC_Reclustered_CCA_Average, features = MAC.top5$gene, group.bar = T, size = 4, angle = 90,draw.lines=F, raster = F) + NoLegend()
heatmac <- DoHeatmap(MAC_Reclustered_scvi_Average, features = MAC.top5$gene, group.bar = T, size = 4, angle = 90,draw.lines=F, raster = F) + NoLegend()
heatmac

```
To make the annotation properly, I am using the paper supplementary table for MPC markers to check the clustering information. FeaturePlot with a set of overlap markers using seurat "AddModuleScore" function with default assay "RNA". Clusters annotations with selected markers checking using the Cochain's integrated scRNASeq paper (ESC, Cardiovascular Research, 2022) Supplementary table for Myeloid clusters cells reclustering table.


```{r, fig.width = 20, fig.height = 20}
MPCs <- read.xlsx("/Users/xz289/Documents/Minoru/Seurat_v3/cvac161_supplementary_data/Supp_xls_1_Mouse_MPC_Marker_Genes.xlsx", sheetIndex=1)
MPCs <- MPCs[-1,]
MPC_clust <- unique(MPCs[-1,7])
addM_MAC  <- MAC_Reclustered
DefaultAssay(addM_MAC) <- "RNA"

Fplt_fn <- function(seobj, Minput, clustM, clustCol, geneCol){
  celltype_marker_gene_list <- list(Minput[Minput[,clustCol]==clustM, geneCol])
  newobject <- AddModuleScore(object = seobj, features = celltype_marker_gene_list, 
                              name = paste0(clustM, "_score"))
  Fplt_M    <- FeaturePlot(object = newobject, features = paste0(clustM, "_score1")) +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
    
  Fplt_M
}
MclustPlt <- list()
for(i in c(1:6,8:15)){
  print(i)
  MclustPlt[[i]] <- Fplt_fn(addM_MAC, MPCs, MPC_clust[i], 7, 8)
  MclustPlt
}

i = 7
celltype_marker_gene_list <- list(MPCs[MPCs[,7]=="CCR2intMHCII+", 8])
newobject <- AddModuleScore(object = addM_MAC, features = celltype_marker_gene_list, 
                              name = "CCR2intMHCII_score")
Fplt_M    <- FeaturePlot(object = newobject, features = "CCR2intMHCII_score1") +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu")))
MclustPlt[[7]] <-  Fplt_M  

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/MPCs_FeaturePlot_clustersMarkers_Res12_c19_June_2023.pdf", width = 16, height = 16)
umapcplt <- DimPlot(MAC_Reclustered, label=T)
plot_grid(MclustPlt[[1]], MclustPlt[[2]], MclustPlt[[3]], MclustPlt[[4]],
          MclustPlt[[5]], MclustPlt[[6]], MclustPlt[[7]], MclustPlt[[8]],
          MclustPlt[[9]], MclustPlt[[10]], MclustPlt[[11]], MclustPlt[[12]],
          MclustPlt[[13]], MclustPlt[[14]], MclustPlt[[15]], umapcplt, ncol = 4, byrow=T)
dev.off()
```



```{r, annotation with global and fine clustering}
new.cluster.ids <- c("Adv/Res", "Adv/Res", "CCR2-MHCII", "Adv/Res",
                     "Inflam", "Mac_AIR/Trem2", "Inflam", "Trem2-Slamf9",
                     "Adv/Res", "IFNIC", "Inflam", "Adv/Res", "Prolif_G2M",
                     "Inflam", "Adv/Res", "Prolif_G2M", "Monocyte", 
                     "Mac_AIR/Trem2", "Mono/DCs")
new.cluster.idsm <- c("Adv/Res", "Adv/Res", "CCR2-MHCII", "Adv/Res",
                     "IFN/Inflam", "Mac_AIR/Trem2", "IFN/Inflam", "Mac_AIR/Trem2",
                     "Adv/Res", "IFN/Inflam", "IFN/Inflam", "Adv/Res", "Prolif_G2M",
                     "IFN/Inflam", "Adv/Res", "Prolif_G2M", "Monocyte", 
                     "Mac_AIR/Trem2", "Mono/DCs")

message("+-------Add the cluster celltype annotations-------+")
## global annotation with 3 main subtype of Macrophages
Idents(MAC_Reclustered) <- MAC_Reclustered[["seurat_clusters"]]
names(new.cluster.idsm) <- levels(MAC_Reclustered)
MAC_Reclustered <- RenameIdents(MAC_Reclustered, new.cluster.idsm)
#MAC_Reclustered[["Global_CCA_Annotation"]] <- Idents(object = MAC_Reclustered)
MAC_Reclustered[["Global_scvi_Annotation"]] <- Idents(object = MAC_Reclustered)

## Fine annotation 6 subcluster of macrophage
Idents(MAC_Reclustered) <- MAC_Reclustered[["seurat_clusters"]]
names(new.cluster.ids) <- levels(MAC_Reclustered)
MAC_Reclustered <- RenameIdents(MAC_Reclustered, new.cluster.ids)
#MAC_Reclustered[["Fine_CCA_Annotation"]] <- Idents(object = MAC_Reclustered)
MAC_Reclustered[["Fine_scvi_Annotation"]] <- Idents(object = MAC_Reclustered)
```
Generate UMAP plot and highlight the Mac_AIR clusters in William's paper.

```{r, fig.width = 12, fig.height = 8}


pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Macrophage_reclustered_Ldlr_ChowHFD_UMAP_MacAir_Res12_C19_Jun_2023.pdf",width = 12, height = 10, onefile = T)

Idents(MAC_Reclustered) <- MAC_Reclustered[["Global_CCA_Annotation"]]
umap_allG  <- UMAPPlot(MAC_Reclustered, label=T) 
Idents(MAC_Reclustered) <- MAC_Reclustered[["Fine_CCA_Annotation"]]
umap_allf   <- UMAPPlot(MAC_Reclustered, label=T)
macair_highlight <- rownames(MAC_Reclustered@meta.data[MAC_Reclustered@meta.data$Williams_C57_ID=="Mac_AIR",])
umap_macair<- UMAPPlot(MAC_Reclustered, label=F, cells.highlight=list(macair_highlight)) 
plot_grid(umap_allG, umap_macair, umap_all, umap_allf, ncol = 2)
dev.off()

```


```{r,fig.width = 5, fig.height = 4 }
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Macrophage_Ldlr_ChowHFD_res12_c19_seldefineColor_UMAP_Global_GlobalF_June_2023.pdf", width = 5, height = 4)
Idents(MAC_Reclustered) <- MAC_Reclustered[["Global_CCA_Annotation"]]
UMAP= UMAPPlot(MAC_Reclustered, label=F)+
  scale_color_manual(values=c("darkred", "darkorange", "darkgray",  "darkorchid", "cyan", "pink", "green"))
LabelClusters(UMAP, id = "ident",  fontface = "bold", size=5)

Idents(MAC_Reclustered) <- MAC_Reclustered[["seurat_clusters"]]
## merge the MacAir and Trem2, split inflam and IFNIC
new.cluster.ids1 <- c("Adv/Res", "Adv/Res", "CCR2-MHCII", "Adv/Res",
                     "Inflam", "Mac_AIR/Trem2", "Inflam", "Mac_AIR/Trem2",
                     "Adv/Res", "IFNIC", "Inflam", "Adv/Res", "Prolif_G2M",
                     "Inflam", "Adv/Res", "Prolif_G2M", "Monocyte", 
                     "Mac_AIR/Trem2", "Mono/DCs")
names(new.cluster.ids1) <- levels(MAC_Reclustered)
MAC_Reclustered <- RenameIdents(MAC_Reclustered, new.cluster.ids1)
MAC_Reclustered[["GlobalF_CCA_Annotation"]] <- Idents(object = MAC_Reclustered)
#Idents(MAC_Reclustered) <- MAC_Reclustered[["GlobalF_CCA_Annotation"]]
UMAPF= UMAPPlot(MAC_Reclustered, label=F)+
  scale_color_manual(values=c("darkred", "darkorange", "darkgray",  "darkorchid","gray85", "cyan", "pink", "green"))
LabelClusters(UMAPF, id = "ident",  fontface = "bold", size=5)

dev.off()
```

```{r, fig.width = 10, fig.height = 4 }

message("+------------------Extended Fig10b---------------------------------------------+")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ExtFig10b_v1_Macrophage_Ldlr_ChowHFD_res12_c19_seldefineColor_UMAP.pdf", width = 9, height = 3.5)
UMAPF1 = UMAPPlot(MAC_Reclustered, label=T)+
  scale_color_manual(values=c("darkred", "darkorange", "darkgray",  "darkorchid","gray85", "cyan", "pink", "green")) 

plot_grid(UMAPF1, umap_macair, ncol = 2)
dev.off()


pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ExtFig10b_v1_Macrophage_Ldlr_ChowHFD_res12_c19_seldefineColor_UMAP_subMac_18_12_2023.pdf", width = 9, height = 3.5)
MAC_Reclustered_sub <- subset(MAC_Reclustered, idents = c("IFN/Inflam", "Adv/Res", "Mac_AIR/Trem2", "CCR2-MHCII"))
UMAPF2= UMAPPlot(MAC_Reclustered_sub, label=T)+
     scale_color_manual(values=c("darkred", "cyan", "darkgray",  "darkorchid"))
#LabelClusters(UMAP, id = "ident",  fontface = "bold", size=5)
umap_macair2<- UMAPPlot(MAC_Reclustered_sub, label=F, cells.highlight=list(macair_highlight)) 
plot_grid(UMAPF2, umap_macair2, ncol = 2)
dev.off()

```
```{r, fig.width = 5, fig.height = 4 }
message("+------------------Extended Fig10c---------------------------------------------+")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ExtFig10c_v1_Macrophage_Ldlr_ChowHFD_res12_c19_seldefineColor_UMAP_GlobalF.pdf", width = 9, height = 9)
UMAP_study = UMAPPlot(MAC_Reclustered, label=F, split.by = "Protocol", ncol = 3)+
  scale_color_manual(values=c("darkred", "darkorange", "darkgray", "darkorchid", "gray85", "cyan","pink", "green"))+theme(legend.position="none")

UMAP_study
dev.off()

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ExtFig10c_v1_Macrophage_Ldlr_ChowHFD_res12_c19_seldefineColor_UMAP_subMac_GlobalF_18_12_2023.pdf", width = 9, height = 9)
UMAP_study = UMAPPlot(MAC_Reclustered_sub, label=F, split.by = "Protocol", ncol = 3)+
  scale_color_manual(values=c("darkred", "cyan", "darkgray", "darkorchid"))+theme(legend.position="none")

UMAP_study
dev.off()

```

DotPlot for papers Fig4d, for mouse with selected genes, with fine-clusters and group-clustered as first version.

```{r, fig.width = 16, fig.height = 6}
DefaultAssay(MAC_Reclustered) <- "RNA"
Mac_genes <- c("Pf4","F13a1","Lyve1","Csf1r","Cd163","Ednrb","Folr2","Mrc1","Sepp1",
               "Nrp1","Timd4","Cd209f", "Cd209d","Phactr1",
               "Cxcl2","Nfkbiz","Nfkbia","Nlrp3","Cd14","Cebpb","Il1b","Il1rn", 
               "Ifit3", "Stat1", "Mnda", "Irf7","Isg15",  "Ifit2" ,  "Ccl12", "Fcgr4",
               "Lgals3", "Trem2","Cd9","Ctsb","Gngt2","Spp1","Acp5","Fabp5","Cadm1",
               "Mmp12", "Itgax","Gpnmb")
## revised and reordered the selected genes. 
Mac_genes_new <- c("Ednrb","Lyve1","Cd163","Pf4","F13a1","Folr2","Nrp1","Timd4","Phactr1",
               "Nfkbiz","Nfkbia","Cxcl2","Nlrp3","Cebpb","Cd14", "Il1b",
               "Mnda","Ifit3", "Stat1", "Ifit2","Isg15","Fcgr4",
               "Lgals3", "Trem2","Cd9","Spp1","Fabp5","Itgax","Mmp12", "Cadm1","Gpnmb")
## due to the updates seruat version 5, the following line is to change the idents.
MAC_Reclustered <- SetIdent(MAC_Reclustered, value = MAC_Reclustered@meta.data$GlobalF_CCA_Annotation)
MAC_Reclustered_subGF  <- subset(MAC_Reclustered, idents=c("Adv/Res", "Inflam", "IFNIC", "Mac_AIR/Trem2"))
levels(MAC_Reclustered_subGF) <- c("Adv/Res", "Inflam", "IFNIC", "Mac_AIR/Trem2")

message("+--------Fig4dvGF: used Adv/Res, Inflam, IFNIC, Mac_AIR/Trem2-------+")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Fig4d_vGF_Mouse_integrated_selMac_dotplot_selMarkers_June_2023_GlobalfineCluster_Res06_C10_GFC4_Nov_2023.pdf", width=10, height=4.0)
plt_dotGF <- DotPlot(MAC_Reclustered_subGF, features=Mac_genes_new, scale = T)+
  RotatedAxis()+scale_color_viridis() +
  scale_size(range = c(2, 7)) + xlab("") + ylab("") +
  theme(legend.position="top",legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(0.5, 'cm'), #change legend key height
        legend.key.width = unit(1.2, 'cm'), #change legend key width
        legend.title = element_text(size=12, face = "bold"), #change legend title font size
        legend.text = element_text(size=10, face = "bold"),
        axis.text.x = element_text(size=12, face = "italic"),
        axis.text.y = element_text(size=12,face = "bold"))
plt_dotGF
dev.off()

message("+--------Fig4dvG: used Adv/Res, IFN/Inflam, Mac_AIR/Trem2-------+")

#Idents(MAC_Reclustered) <- MAC_Reclustered[["Global_CCA_Annotation"]]
MAC_Reclustered <- SetIdent(MAC_Reclustered, value = MAC_Reclustered@meta.data$Global_CCA_Annotation)
MAC_Reclustered_sub <- subset(MAC_Reclustered, idents=c("Adv/Res", "IFN/Inflam", "Mac_AIR/Trem2"))
levels(MAC_Reclustered_sub) <- c("Adv/Res", "IFN/Inflam", "Mac_AIR/Trem2")
## need to reorder the genes

message("+------------------Figure4d---------------------------------------------+")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Fig4d_vG_Mouse_integrated_selMac_dotplot_selMarkers_Nov_2023_MerCluster_Res12_C19_MC3_Nov_2023.pdf", width=10, height=3)
Mac_genes_neword <- c("Ednrb","Lyve1","Timd4","Pf4", "Cd163","F13a1","Folr2","Nrp1","Phactr1",
                      "Isg15","Stat1","Mnda","Ifit3", "Ifit2",
                      "Nfkbiz","Nfkbia","Cxcl2","Nlrp3","Cebpb","Cd14", "Il1b","Fcgr4",
                      "Lgals3", "Trem2","Cd9","Spp1","Fabp5","Itgax","Mmp12", "Cadm1","Gpnmb")
plt_dotG <- DotPlot(MAC_Reclustered_sub, features=Mac_genes_neword, scale = T)+
  RotatedAxis()+scale_color_viridis() +
  scale_size(range = c(2, 7)) + xlab("") + ylab("") +
  theme(legend.position="top",legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(0.5, 'cm'), #change legend key height
        legend.key.width = unit(1.2, 'cm'), #change legend key width
        legend.title = element_text(size=12, face = "bold"), #change legend title font size
        legend.text = element_text(size=10, face = "bold"),
        axis.text.x = element_text(size=12, face = "italic"),
        axis.text.y = element_text(size=12,face = "bold"))
plt_dotG
dev.off()

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Fig4d_vG_Mouse_integrated_selMac_dotplot_selMarkers_Nov_2023_MerCluster_Res12_C19_MC3_Dec_2023.pdf", width=10, height=3.5)
Mac_genes_neword_1 <- c("Ednrb","Lyve1","Timd4","Pf4", "Cd163","F13a1","Folr2","Nrp1","Phactr1", 
                   "Lgals3", "Trem2","Cd9","Spp1","Fabp5","Itgax","Mmp12", "Cadm1","Gpnmb",
                      "Isg15","Stat1","Mnda","Ifit3", "Ifit2",
                      "Nfkbiz","Nfkbia","Cxcl2","Nlrp3","Cebpb","Cd14", "Il1b","Fcgr4")
Idents(MAC_Reclustered_sub) <- factor(Idents(MAC_Reclustered_sub), levels = c("IFN/Inflam", "Mac_AIR/Trem2", "Adv/Res"))
plt_dotG1 <- DotPlot(MAC_Reclustered_sub, features=Mac_genes_neword_1, scale = T)+
  RotatedAxis()+scale_color_viridis() +
  scale_size(range = c(2, 7)) + xlab("") + ylab("") +
  theme(legend.position="top",legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(0.5, 'cm'), #change legend key height
        legend.key.width = unit(1.2, 'cm'), #change legend key width
        legend.title = element_text(size=12, face = "bold"), #change legend title font size
        legend.text = element_text(size=10, face = "bold"),
        axis.text.x = element_text(size=12, face = "italic"),
        axis.text.y = element_text(size=12,face = "bold"))
plt_dotG1
dev.off()

```
Next to perform some DEGs analysis and pathway analysis. <br>
1) Nrp1 high vs Nrp1 low, pathway analysis;<br>
2) Reslike, MacAir/Trem2, Inflam, IFNIC HFDvsCtrl DEGs;<br>
3) GO pathway analysis for 2) to plot unique and common GOs (mainly in Biological Process); <br>

```{r, define the different conditions}
## 1) Ctrl HFD, Condition
testall <- MAC_Reclustered
Idents(testall) <- testall@meta.data$Protocol
testall@meta.data$Condition <- ifelse(testall@meta.data$Protocol=="Cochain_Chow"|
                                      testall@meta.data$Protocol=="Williams_C57", "Chow", "HFD")
## 2) assign different Conditons with cell type specific
## global condition
#testall@meta.data$GlobalCondition <- paste0(testall@meta.data$Global_CCA_Annotation, "_",
                                            testall@meta.data$Condition)
#testall@meta.data$GlobalCondition <- paste0(testall@meta.data$Global_scvi_Annotation, "_",
                                            testall@meta.data$Condition)
table(testall@meta.data$GlobalCondition)

## globalF condition (split inflam and IFNIC)
#testall@meta.data$GlobalFCondition <- paste0(testall@meta.data$GlobalF_CCA_Annotation, "_",
                                            testall@meta.data$Condition)
testall@meta.data$GlobalFCondition <- paste0(testall@meta.data$GlobalF_scvi_Annotation, "_",
                                            testall@meta.data$Condition)
table(testall@meta.data$GlobalFCondition)

## 3) split by time (3wk, 10+11wks, 20 week for HFD) and add celltype
timeCond <- ifelse(testall@meta.data$Protocol=="Cochain_Ldlr_20_weeks_HFD", "wk20_HFD",
                   "wk11_HFD")
timeCond <- ifelse(testall@meta.data$Protocol=="Williams_C57" | testall@meta.data$Protocol=="Cochain_Chow", "Ctrl", timeCond)
timeCond <- ifelse(testall@meta.data$Protocol=="Williams_21days_HFD", "wk3_HFD", timeCond)
table(timeCond)
testall@meta.data$TimeCond <- timeCond

#testall@meta.data$NewTimeCond <- paste0(testall@meta.data$Global_CCA_Annotation, "_",timeCond)
testall@meta.data$NewTimeCond <- paste0(testall@meta.data$Global_scvi_Annotation, "_",timeCond)
table(testall@meta.data$NewTimeCond)

#testall@meta.data$NewFTimeCond <- paste0(testall@meta.data$GlobalF_CCA_Annotation, "_",timeCond)
testall@meta.data$NewFTimeCond <- paste0(testall@meta.data$GlobalF_scvi_Annotation, "_",timeCond)
table(testall@meta.data$NewFTimeCond)

MAC_Reclustered <- testall

message("+---- Final Seurat object for mouse macrophage reclustering---------------------------------+")
save(MAC_Reclustered, file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/NewMAC_Reclustered_withConditions_meta_Robj_Res12_C19_GC3_GFC4_June_2023.RData")
message("+-------------------------------------------------------------------------------------------+")
```
Next we will perform the DEGs analysis as following. <br>
\item HFDvsCtrl Adv/Res, IFN, Inflam, Mac_AIR/Trem2; <br>
\item HFDvsCtrl across time points <br>
From the paper "Bias, robustness and scalability in single-cell differential expression analysis" by Soneson and Robinson (2018, https://www.nature.com/articles/nmeth.4612, DOI:: http://dx.doi.org/10.1038/nmeth.4612), I will use Figure 5 summary, choose the "MAST" method which shows the better performance across different method and with flexible modelling. And also apply the other R package 'variancePartition' (which also called dream) to perform modelling analysis.(https://bioconductor.org/packages/release/bioc/vignettes/variancePartition/inst/doc/dream.html) Both method are not only use the design between groups. <br>

\item 1) Seurat, Wilcox, DESeq2 and MAST; <br>
\item 2) Dream, model  treatment(HFD, Chow) with/without model different study too. For small datasets, the Kenward-Roger method can be more powerful. But it is substantially more computationally intensive. <br>


```{r, use the Adv/Res group to test the methods}

DefaultAssay(MAC_Reclustered) <- "RNA"
#Idents(MAC_Reclustered) <- MAC_Reclustered[["GlobalCondition"]]
Idents(MAC_Reclustered) <- "GlobalCondition"
# 1) seurat default findmarkers wilcox, DESeq2
Res_wilcox <- FindMarkers(MAC_Reclustered, assay = "RNA", ident.1 = "Adv/Res_HFD", ident.2 = "Adv/Res_Chow", 
                          verbose = FALSE, logfc.threshold = 0.00) 
Res_DEseq2 <- FindMarkers(MAC_Reclustered, assay = "RNA", ident.1 = "Adv/Res_HFD", ident.2 = "Adv/Res_Chow", 
                          verbose = FALSE, logfc.threshold = 0.00, test.use = "DESeq2")
Res_MAST   <- FindMarkers(MAC_Reclustered, assay = "RNA", ident.1 = "Adv/Res_HFD", ident.2 = "Adv/Res_Chow", 
                          verbose = FALSE, logfc.threshold = 0.00, test.use = "MAST")

# 2) DREAM
library('variancePartition')
library('edgeR')
library('BiocParallel')
param = SnowParam(4, "SOCK", progressbar=TRUE)

Res_sub <- subset(MAC_Reclustered, idents = c("Adv/Res_Chow", "Adv/Res_HFD"))
Resmatrix <- GetAssayData(Res_sub)
# Standard usage of limma/voom
Res_dge = DGEList(Resmatrix)
Res_dge = calcNormFactors(Res_dge)
Res_meta = Res_sub@meta.data
# The variable to be tested must be a fixed effect
Gform  <- ~ GlobalCondition + (1|Protocol) 

# estimate weights using linear mixed model of dream
Res_vobjDream = voomWithDreamWeights(Res_dge, Gform, Res_meta, BPPARAM=param)
# Fit the dream model on each gene
# By default, uses the Satterthwaite approximation for the hypothesis test
Res_fitmm = dream(Res_vobjDream, Gform, Res_meta )
Res_fitmm = eBayes(Res_fitmm)
Res_tab <- topTable(Res_fitmm, n = dim(Res_fitmm)[1])
head(Res_tab)
# no significant genes identified
message("+---just use treatment as model design--------+")
G1form <- ~ GlobalCondition
Res_vobjDream1 = voomWithDreamWeights(Res_dge, G1form, Res_meta, BPPARAM=param)
Res_fitmm1 = dream(Res_vobjDream1, G1form, Res_meta )
Res_fitmm1 = eBayes(Res_fitmm1)
Res_tab1 <- topTable(Res_fitmm1, n = dim(Res_fitmm1)[1])
head(Res_tab1)
```
To test the different methods performance, overlap DEGs will be performed with padj_sig less than 0.05 and absolute fold change value is 1.2. UpsetR will be presented. Here, we split up and down regulated separate to perform the overlap consistency across different methods.

```{r, fig.width = 7, fig.height = 5}
psig <- 0.05; l2fcsig <- 0.27 # 1.2-foldChange
colnames(Res_tab1) <- c("avg_log2FC", "AveExpr", "t", "p_val", "p_val_adj", "B")
Res_ComList <- list(Res_wilcox, Res_DEseq2, Res_MAST, Res_tab1)
Res_ComsigL <- lapply(Res_ComList, function(x) x[x$p_val_adj<=psig&abs(x$avg_log2FC)>=l2fcsig, ])
lapply(Res_ComsigL, dim)
Res_upgenesL    <- unique(unlist(lapply(Res_ComsigL, function(x) rownames(x[x$avg_log2FC>0,]))))
Res_dwgenesL    <- unique(unlist(lapply(Res_ComsigL, function(x) rownames(x[x$avg_log2FC<0,]))))
Res_upgenesL <- Res_upgenesL[order(Res_upgenesL)]
Res_dwgenesL <- Res_dwgenesL[order(Res_dwgenesL)]

upover_upsetR <- matrix(0, ncol = 4, nrow=length(Res_upgenesL))
dwover_upsetR <- matrix(0, ncol = 4, nrow=length(Res_dwgenesL))

for(i in 1:4){
  upover_upsetR[,i] <- ifelse(Res_upgenesL%in%rownames(Res_ComsigL[[i]][Res_ComsigL[[i]]$avg_log2FC>0,])==T,
                              1, 0)
  dwover_upsetR[,i] <- ifelse(Res_dwgenesL%in%rownames(Res_ComsigL[[i]][Res_ComsigL[[i]]$avg_log2FC<0,])==T,
                              1, 0)
}

rownames(upover_upsetR) <- Res_upgenesL; colnames(upover_upsetR) <- c("Wilcox", "DESeq2", "MAST", "DREAM")
rownames(dwover_upsetR) <- Res_dwgenesL; colnames(dwover_upsetR) <- c("Wilcox", "DESeq2", "MAST", "DREAM")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_UpsetR_June_2023.pdf", width=7, height=5)
UpSetR::upset(as.data.frame(upover_upsetR), order.by = "freq")
UpSetR::upset(as.data.frame(dwover_upsetR), order.by = "freq")
dev.off()

rownames(upover_upsetR[rowSums(upover_upsetR)==4,]) ##  
rownames(dwover_upsetR[rowSums(dwover_upsetR)==4,])

write.xlsx(Res_wilcox, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.xlsx", append = F, sheetName = "Res_Wilcox")
write.xlsx(Res_DEseq2, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_DESeq2")
write.xlsx(Res_MAST, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_MAST")
write.xlsx(Res_tab1, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_Dream")
write.xlsx(Res_tab, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_DreamSample")

```
After the above analysis, I would like to use "MAST" to carry on the following comparison analysis to catch most of significant DEGs and perform the enrichment analysis too. The DREAM with modelling added the protocol which makes no significant DEGs, but the direction of the Adv/Res key markers are down regulated. <br>

Following analysis will be performed for IFN/Inflam and Mac_AIR/Trem2, Mac_AIR/Trem2 has more number of cells in HFD compared to Chow. The other categories the number of cells are similar.

```{r}
IFNI_MAST   <- FindMarkers(MAC_Reclustered, assay = "RNA", ident.1 = "IFN/Inflam_HFD", ident.2 = "IFN/Inflam_Chow", 
                          verbose = FALSE, logfc.threshold = 0.00, test.use = "MAST")

AIRT_MAST   <- FindMarkers(MAC_Reclustered, assay = "RNA", ident.1 = "Mac_AIR/Trem2_HFD", ident.2 =
                           "Mac_AIR/Trem2_Chow", verbose = FALSE, logfc.threshold = 0.00, test.use = "MAST")

write.xlsx(Res_MAST, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_MAST")
write.xlsx(IFNI_MAST, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "IFN_Inflam_MAST")
write.xlsx(AIRT_MAST, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DDEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "MacAIR_Trem2_MAST")
```
Perform the DEGs analysis across the time for global annotation with 3 clusters for Macrophages.

```{r}
Idents(MAC_Reclustered) <- MAC_Reclustered[["NewTimeCond"]]
Timelab  <- c("wk3", "wk11", "wk20")
AdvR_labs <- cbind(paste0("Adv/Res_",Timelab,"_HFD"), rep("Adv/Res_Ctrl",3))
IFNI_labs <- cbind(paste0("IFN/Inflam_",Timelab,"_HFD"), rep("IFN/Inflam_Ctrl",3))
AIRT_labs <- cbind(paste0("Mac_AIR/Trem2_",Timelab,"_HFD"), rep("Mac_AIR/Trem2_Ctrl",3))

AdvR_Time_MAST   <- apply(AdvR_labs, 1, function(x) FindMarkers(MAC_Reclustered, assay = "RNA", 
                                                                ident.1 = x[1], ident.2 = x[2],
                                                                verbose = FALSE, logfc.threshold = 0.00,
                                                                test.use = "MAST"))
IFNI_Time_MAST   <- apply(IFNI_labs, 1, function(x) FindMarkers(MAC_Reclustered, assay = "RNA", 
                                                                ident.1 = x[1], ident.2 = x[2],
                                                                verbose = FALSE, logfc.threshold = 0.00,
                                                                test.use = "MAST"))

AIRT_Time_MAST   <- apply(AIRT_labs, 1, function(x) FindMarkers(MAC_Reclustered, assay = "RNA", 
                                                                ident.1 = x[1], ident.2 = x[2],
                                                                verbose = FALSE, logfc.threshold = 0.00,
                                                                test.use = "MAST"))

save(AdvR_Time_MAST , AIRT_MAST, AIRT_Time_MAST, IFNI_MAST, IFNI_Time_MAST, Res_MAST, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.RData")

```
Next, we will generate volcano plots and heatmaps for selected MacGenes across the three macrophage types with log2FoldChange. Heatmaps first across the subtypes and then across the time with subtypes.

```{r, fig.width= 5, fig.height = 10}
macheatGenes <- c(Mac_genes, "Igfbp4", "Igf1", "Igf1r")

Res_Ghmat1 <- Res_MAST[rownames(Res_MAST)%in%macheatGenes, c("avg_log2FC", "p_val_adj")]
Res_Ghmat1$gene <- rownames(Res_Ghmat1)
IFN_Ghmat1 <- IFNI_MAST[rownames(IFNI_MAST)%in%macheatGenes, c("avg_log2FC", "p_val_adj")]
IFN_Ghmat1$gene <- rownames(IFN_Ghmat1)
AIR_Ghmat1 <- AIRT_MAST[rownames(AIRT_MAST)%in%macheatGenes, c("avg_log2FC", "p_val_adj")]
AIR_Ghmat1$gene <- rownames(AIR_Ghmat1)
macheat_dat <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T),
        list(Res_Ghmat1, IFN_Ghmat1, AIR_Ghmat1))

macheat_pltd <- macheat_dat[, c(2,4,6)]
rownames(macheat_pltd) <- macheat_dat$gene
colnames(macheat_pltd) <- c("Adv_Res", "IFN_Inflam", "MacAIR_Trem2")

library(circlize)
library(grid)
colsf1 = colorRamp2( c(-5,-2, 0, 2, 5), c("darkgreen", "green3", "gray95", "slateblue", "darkorchid4"), 
                     space = "RGB") 

macheat_pltheat <- ComplexHeatmap:: Heatmap(as.matrix(macheat_pltd), na_col ="black",
                            name = "macheat",  #col = colsf1,
                           row_title = "", column_title = NULL, 
                           show_row_names = TRUE, show_column_names = TRUE,
                           heatmap_legend_param = list(title = "log2FoldChange", 
                                                       legend_height = unit(6, "cm"), 
                                                       title_position = "leftcenter-rot"), 
                           cluster_columns = FALSE, row_names_side ="right", 
                           cluster_rows = FALSE, 
                           column_names_rot = 0, column_names_side = "bottom", 
                           column_title_rot = 0, column_names_centered = TRUE,
                           column_names_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_names_gp = gpar(fontsize = 10, fontface = "bold.italic"),
                           column_title_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_title_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_title_rot = 0,
                           row_gap = unit(2, "mm"), column_gap = unit(3, "mm")#,
                           #width = ncol(macheat_pltd)*unit(10, "mm"), 
                           #height = nrow(macheat_pltd)*unit(4, "mm")
                           )
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Heatmap_DEGs_MAST_GlobalC3_selMarkers_June_2023.pdf")
macheat_pltheat
dev.off()

write.xlsx(AdvR_Time_MAST[[1]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_wk3")
write.xlsx(AdvR_Time_MAST[[2]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_wk11")
write.xlsx(AdvR_Time_MAST[[3]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "Res_wk20")

write.xlsx(IFNI_Time_MAST[[1]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "IFN_wk3")
write.xlsx(IFNI_Time_MAST[[2]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "IFN_wk11")
write.xlsx(IFNI_Time_MAST[[3]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "IFN_wk20")

write.xlsx(AIRT_Time_MAST[[1]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "AIR_wk3")
write.xlsx(AIRT_Time_MAST[[2]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "AIR_wk11")
write.xlsx(AIRT_Time_MAST[[3]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_MACs_HFDvsCtrl_MAST_summaryTable_June_2023.xlsx", append = T, sheetName = "AIR_wk20")

```

```{r, fig.width= 25, fig.height = 10}
hpltm_datasort <- function(datinput, mgenes, selcolName){
  subd <- datinput[rownames(datinput)%in%mgenes, selcolName]
  subd$gene <- rownames(subd)
  subd
}
selcolName <- c("avg_log2FC", "p_val_adj")
mgenes <- macheatGenes
Res_GhmatT <- lapply(AdvR_Time_MAST,  function(x) hpltm_datasort(x, mgenes, selcolName))
IFN_GhmatT <- lapply(IFNI_Time_MAST,  function(x) hpltm_datasort(x, mgenes, selcolName))
AIR_GhmatT <- lapply(AIRT_Time_MAST,  function(x) hpltm_datasort(x, mgenes, selcolName))

Resheat_datT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), Res_GhmatT)[,c(1,2,4,6)]
IFNheat_datT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), IFN_GhmatT)[,c(1,2,4,6)]
AIRheat_datT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), AIR_GhmatT)[,c(1,2,4,6)]

colnames(Resheat_datT)[2:4] <- paste0("Res_w", c(3,11,20))
colnames(IFNheat_datT)[2:4] <- paste0("IFN_w", c(3,11,20))
colnames(AIRheat_datT)[2:4] <- paste0("AIR_w", c(3,11,20))

TimeMAC_mer  <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), 
                       list(Resheat_datT, IFNheat_datT, AIRheat_datT))

TimeMAC_mer1 <- TimeMAC_mer[,-1]
rownames(TimeMAC_mer1) <- TimeMAC_mer$gene

macheat_pltheatT_v1 <- ComplexHeatmap:: Heatmap(as.matrix(TimeMAC_mer1), na_col ="black",
                           name = "macheatTime",  #col = colsf1,
                           row_title = "", column_title = NULL, 
                           show_row_names = TRUE, show_column_names = TRUE,
                           heatmap_legend_param = list(title = "L2FC(HFDvsChow)", 
                                                       legend_height = unit(6, "cm"), 
                                                       title_position = "leftcenter-rot"), 
                           cluster_columns = FALSE, row_names_side ="right", 
                           cluster_rows = FALSE, 
                           column_split = rep(c("Adv/Res", "IFN/Inflam", "MacAIR/Trem2"), each=3),
                           column_labels = rep(c("3", "11", "20"), length=9),
                           column_names_rot = 0, column_names_side = "bottom", 
                           column_title_rot = 0, column_names_centered = TRUE,
                           column_names_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_names_gp = gpar(fontsize = 10, fontface = "bold.italic"),
                           column_title_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_title_gp = gpar(fontsize = 10, fontface = "bold", col ),
                           row_title_rot = 0, column_gap = unit(3, "mm"))
TimeMAC_mer2 <- TimeMAC_mer1[, c(1, 4, 7, 2, 5, 8, 3, 6, 9)]
macheat_pltheatT_v2 <- ComplexHeatmap:: Heatmap(as.matrix(TimeMAC_mer1), na_col ="black",
                           name = "macheatTime",  #col = colsf1,
                           row_title = "", column_title = NULL, 
                           show_row_names = TRUE, show_column_names = TRUE,
                           heatmap_legend_param = list(title = "log2FoldChange", 
                                                       legend_height = unit(6, "cm"), 
                                                       title_position = "leftcenter-rot"), 
                           cluster_columns = FALSE, row_names_side ="right", 
                           cluster_rows = FALSE, 
                           column_split = rep(c("3", "11", "20"), each=3),
                           column_labels = rep(c("Res", "IFN", "AIR"), length=9),
                           column_names_rot = 0, column_names_side = "bottom", 
                           column_title_rot = 0, column_names_centered = TRUE,
                           column_names_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_names_gp = gpar(fontsize = 10, fontface = "bold.italic"),
                           column_title_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_title_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_title_rot = 0, column_gap = unit(3, "mm"))
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Heatmap_DEGs_MAST_GlobalC3_Time_selMarkers_June_2023.pdf")
macheat_pltheatT_v1
macheat_pltheatT_v2
dev.off()

```
After meeting with Ziad (30/06/2023) the heatmap will be generated for time difference by the following <br>
\item For each macrophage subtype only choose the surface markers <br>
\item how to present the res, ifn and air all? plot the average normalised counts for hfd and chow. <br>

```{r, heatmap}
Res_markers <- c("Pf4","F13a1","Lyve1","Csf1r","Cd163","Ednrb","Folr2","Mrc1","Sepp1",
               "Nrp1","Timd4","Cd209f", "Cd209d", "Igfbp4", "Igf1", "Igf1r")
IFN_markers <- c("Cxcl2","Nfkbiz","Nfkbia","Nlrp3","Cd14","Cebpb","Il1b","Il1rn", 
               "Stat1", "Mnda", "Irf7","Isg15",  "Ccl12", "Fcgr4")
AIR_markers <- c("Lgals3", "Trem2","Cd9","Ctsb","Spp1","Fabp5","Slamf9", "Maf", "Acp5",
                 "Gngt2", "Mpeg1", "Cd72", "Gpnmb", "Ctsd", "H2-Eb1", "H2-Aa", "Cadm1", "Mmp12")

selcolName <- c("avg_log2FC", "p_val_adj")

Res_selhmatT <- lapply(AdvR_Time_MAST,  function(x) hpltm_datasort(x, Res_markers, selcolName))
IFN_selhmatT <- lapply(IFNI_Time_MAST,  function(x) hpltm_datasort(x, IFN_markers, selcolName))
AIR_selhmatT <- lapply(AIRT_Time_MAST,  function(x) hpltm_datasort(x, AIR_markers, selcolName))

Resheat_seldatT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), Res_selhmatT)[,c(1,2,4,6)]
IFNheat_seldatT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), IFN_selhmatT)[,c(1,2,4,6)]
AIRheat_seldatT <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "gene", all=T), AIR_selhmatT)[,c(1,2,4,6)]

colnames(Resheat_seldatT)[2:4] <- paste0(c(3,11,20), "wks") 
colnames(IFNheat_seldatT)[2:4] <- paste0(c(3,11,20), "wks")
colnames(AIRheat_seldatT)[2:4] <- paste0(c(3,11,20), "wks")

Resheat_seldatT1 <- Resheat_seldatT[,-1]; rownames(Resheat_seldatT1) <- Resheat_seldatT$gene
IFNheat_seldatT1 <- IFNheat_seldatT[,-1]; rownames(IFNheat_seldatT1) <- IFNheat_seldatT$gene
AIRheat_seldatT1 <- AIRheat_seldatT[,-1]; rownames(AIRheat_seldatT1) <- AIRheat_seldatT$gene

all_seldatT1  <- rbind(Resheat_seldatT1, IFNheat_seldatT1, AIRheat_seldatT1)
all_seldatT1$Time <- rep(c("Res", "IFN", "AIR"), c(dim(Resheat_seldatT1)[1],
                                                     dim(IFNheat_seldatT1)[1],
                                                     dim(AIRheat_seldatT1)[1]))


## add extra column of heatmap for overall Res/IFN/AIR key markers log2FoldChange merged one.
Res_noT <- Res_MAST[rownames(Res_MAST)%in%Res_markers, ]
IFN_noT <- IFNI_MAST[rownames(IFNI_MAST)%in%IFN_markers, ]
AIR_noT <- AIRT_MAST[rownames(AIRT_MAST)%in%AIR_markers, ]

all_noT <- rbind(Res_noT, IFN_noT, AIR_noT)[, c("avg_log2FC", "p_val_adj")]

## merge all data and sort by sub macrophage type. 
allhpt_seldat <- merge(all_seldatT1, all_noT, by ="row.names")
allhpt_seldat <- allhpt_seldat[order(allhpt_seldat$Time),]
allhpt_selplt <- allhpt_seldat[,-c(1,5,7)]
colnames(allhpt_selplt) <- c("wkA", "wkB", "wkC", "wkD")
rownames(allhpt_selplt) <- allhpt_seldat[,1] 

colOrd <- factor(colnames(Resheat_seldatT1), levels = c("3wks",  "11wks", "20wks"))
hplt_res_all <- ComplexHeatmap:: Heatmap(as.matrix(allhpt_selplt), na_col ="black",
                           name = "macheatTimeres",  #col = colsf1,
                           row_title = "", column_title = NULL, 
                           show_row_names = TRUE, show_column_names = TRUE,
                           heatmap_legend_param = list(title = "log2FoldChange", 
                                                       legend_height = unit(6, "cm"), 
                                                       title_position = "leftcenter-rot"), 
                           cluster_columns = FALSE, row_names_side ="right", 
                           cluster_rows = TRUE, column_split = c("wkA", "wkB", "wkC", "wkD"),
                           #row_order = c(1:dim(all_seldatT1)[1]),
                           row_split = rep(c("IFN/Inflam", "Mac_AIR/Trem2","Adv/Res"), 
                                           c(dim(AIRheat_seldatT1)[1],dim(IFNheat_seldatT1)[1],
                                             dim(Resheat_seldatT1)[1])),
                           cluster_row_slices = T,
                           column_labels = c("3 weeks", "11 weeks", "20 weeks", "All"),
                           column_order = c(1,2,3,4),
                           column_names_rot = 0, column_names_side = "bottom", 
                           column_title_rot = 0, column_names_centered = TRUE,
                           column_names_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_names_gp = gpar(fontsize = 10, fontface = "bold.italic"),
                           column_title_gp = gpar(fontsize = 12, fontface = "bold"),
                           row_title_gp = gpar(fontsize = 12, fontface = "bold"),
                           row_title_rot = 90, row_gap = unit(3, "mm"), 
                           column_gap = unit(2, "mm"))


pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Heatmap_DEGs_MAST_GlobalC3_acrossTime_selMarkers_July_2023.pdf", height = 7, width = 5)
hplt_res_all
dev.off()

## Remove the all, only keep wk3, 11 and 20. revised on 03112023.
## revised on08/12/2023, change the order of the subtype, Top to bottom: Res/MAcAIR/IFN.
allhptord_selplt <- allhpt_selplt[c( 33:48,  1:18, 19:32),]
splitrow_var <- factor(rep(c( "Adv/Res", "Mac_AIR/Trem2", "IFN/Inflam"), 
                    c(dim(Resheat_seldatT1)[1],dim(AIRheat_seldatT1)[1],dim(IFNheat_seldatT1)[1])),
                    level = c( "Adv/Res", "Mac_AIR/Trem2", "IFN/Inflam"))
                    
hplt_res_wks <- ComplexHeatmap:: Heatmap(as.matrix(allhptord_selplt[,-4]), na_col ="black",
                           name = "macheatTimeres",  #col = colsf1,
                           row_title = "", column_title = NULL, 
                           show_row_names = TRUE, show_column_names = TRUE,
                           heatmap_legend_param = list(title = "log2FoldChange", 
                                                       legend_height = unit(6, "cm"), 
                                                       title_position = "leftcenter-rot"), 
                           cluster_columns = FALSE, row_names_side ="right", 
                           column_split = c("wkA", "wkB", "wkC"),
                           row_order = c(1:dim(all_seldatT1)[1]),
                           row_split = splitrow_var,
                           cluster_row_slices = FALSE, cluster_rows=T,
                           show_row_dend = F, 
                           column_labels = c("3 weeks", "11 weeks", "20 weeks"),
                           column_order = c(1,2,3),
                           column_names_rot = 0, column_names_side = "bottom", 
                           column_title_rot = 0, column_names_centered = TRUE,
                           column_names_gp = gpar(fontsize = 10, fontface = "bold"),
                           row_names_gp = gpar(fontsize = 10, fontface = "bold.italic"),
                           column_title_gp = gpar(fontsize = 12, fontface = "bold"),
                           row_title_gp = gpar(fontsize = 12, fontface = "bold"),
                           row_title_rot = 90, row_gap = unit(3, "mm"), 
                           column_gap = unit(2, "mm"))

message("+----- Addition Figure7a for HFDvsCtrl across time heatmap-----------------------------+")
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Heatmap_DEGs_MAST_GlobalC3_acrossTime_noall_selMarkers_Dec_2023.pdf", height = 7, width = 5)
hplt_res_wks
dev.off()
```


Gene ontology enrichment analysis for HFDvsCtrl on C3 subtype of Macrophages, with/without time line.

```{r}
library(clusterProfiler)
library(org.Mm.eg.db)
load("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/DEGs_methodCompare_MACs_ResAdv_HFDvsCtrl_Wilcox_DESeq2_MAST_DREAM_summaryTable_June_2023.RData")
load("/Users/xz289/Documents/Minoru/Project_mt709_0001/Data/References_Data/GRCm39/Ensembl_GRCm39_entrezID_ensembID_exterName.RData")

GODatasort_fn <- function(dinput, ensE){
  dinput$external_gene_name <- rownames(dinput)
  dinput_mer <- merge(dinput, ensE, by = "external_gene_name")
  goinput <- dinput_mer[,c("entrezgene_id", "ensembl_gene_id", "external_gene_name", "avg_log2FC", "p_val_adj")]
  colnames(goinput) <- c("ENTREZID", "ENSEMBL", "SYMBOL", "L2FC", "Padj")
  goinput
}
ensE <- ensEMBL2id.GO; 
GOinput_ResG3 <- GODatasort_fn(Res_MAST, ensE); 
GOinput_IFNG3 <- GODatasort_fn(IFNI_MAST, ensE);
GOinput_AIRG3 <- GODatasort_fn(AIRT_MAST, ensE);

GOinput_ResG3T <- lapply(AdvR_Time_MAST, function(x) GODatasort_fn(x, ensE))
GOinput_IFNG3T <- lapply(IFNI_Time_MAST, function(x) GODatasort_fn(x, ensE))
GOinput_AIRG3T <- lapply(AIRT_Time_MAST, function(x) GODatasort_fn(x, ensE))

allGOinputList <- list(GOinput_ResG3, GOinput_ResG3T[[1]], GOinput_ResG3T[[2]], GOinput_ResG3T[[3]],
                       GOinput_IFNG3, GOinput_IFNG3T[[1]], GOinput_IFNG3T[[2]], GOinput_IFNG3T[[3]],
                       GOinput_AIRG3, GOinput_AIRG3T[[1]], GOinput_AIRG3T[[2]], GOinput_AIRG3T[[3]])
selSIgDEGs_fn  <- function(einput, psig, l2fcsig){
  einput_sig <- subset(einput, Padj <= psig & abs(L2FC) >= l2fcsig)
  einput_sig <- einput_sig[order(-einput_sig$L2FC), ]
  einput_sig
}
allGOinputDL   <- lapply(allGOinputList, function(x) selSIgDEGs_fn(x, psig=0.05, l2fcsig=0.27))

allSheetNames <- paste0(rep(c("Res", "IFN", "AIR"), each = 4), "_", rep(c("all", "wk3", "wk11", "wk20"), length = 12))
write.xlsx(allGOinputDL[[1]], file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/MAST_DEGs_all_p05_fc1.2_summaryT_July_2023.xlsx",
           append = F, sheetName = allSheetNames[1])
for(i in 2:length(allSheetNames)){
  gc()
  write.xlsx(allGOinputDL[[i]], file =
               "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/MAST_DEGs_allGOinput_p05_fc1.2_summaryT_July_2023.xlsx", 
             append = T, sheetName = allSheetNames[i])
  gc()
}


EnrichGO_fn <- function(einput, psig, l2fcsig, gofun="BP", pcut=0.05){
  einput_sig <- subset(einput, Padj <= psig & abs(L2FC) >= l2fcsig)
  einput_sig <- einput_sig[order(-einput_sig$L2FC), ]
  einput_bkg <- einput[order(-einput$L2FC), ]
  genes      <- einput_sig$ENTREZID[!is.na(einput_sig$ENTREZID)]
  print(dim(einput_sig))
  bkgenes    <- einput_bkg$ENTREZID[!is.na(einput_bkg$ENTREZID)]
  GO_outB    <- enrichGO(gene = genes,
                universe      = bkgenes,
                OrgDb         = org.Mm.eg.db,
                ont           = gofun,
                pAdjustMethod = "BH",
                pvalueCutoff  = pcut,
                readable = T)
  
  GO_outB
  
}

 
GORes_BP   <- EnrichGO_fn(GOinput_ResG3, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05)
GOIFN_BP   <- EnrichGO_fn(GOinput_IFNG3, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05)
GOAIR_BP   <- EnrichGO_fn(GOinput_AIRG3, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05)

GORes_BPT  <- lapply(GOinput_ResG3T, function(x) EnrichGO_fn(x, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05))
GOIFN_BPT  <- lapply(GOinput_IFNG3T, function(x) EnrichGO_fn(x, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05))
GOAIR_BPT  <- lapply(GOinput_AIRG3T, function(x) EnrichGO_fn(x, psig=0.05, l2fcsig=0.27, gofun="BP", pcut=0.05))

save(GORes_BP, GORes_BPT, GOIFN_BP, GOIFN_BPT, GOAIR_BP, GOAIR_BPT, file="/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_output_Res_IFN_AIR_allandTime.RData")

```
Check the number of overlap geneontology biological process across three subtypes (all), and also the same sub-type across the time (3, 11, 20 weeks).

```{r}
checkOverlap_fn <- function(inputDatList){
  overIDs <- unique(unlist(lapply(inputDatList, function(x) x$ID)))
  overmat <- matrix(0, ncol = length(inputDatList), nrow = length(overIDs))
  for(i in 1:length(inputDatList)){
    overmat[,i] <- ifelse(overIDs%in%inputDatList[[i]]$ID==T, 1, 0)
  }
  rownames(overmat) <- overIDs
  overmat
}
subAllList <- list(GORes_BP, GOIFN_BP, GOAIR_BP)
AlloverTab <- checkOverlap_fn(subAllList);
TimeResoverTab <- checkOverlap_fn(GORes_BPT);
TimeIFNoverTab <- checkOverlap_fn(GOIFN_BPT);
TimeAIRoverTab <- checkOverlap_fn(GOAIR_BPT);
```
GO Semantic Similarity Analysis will be performed to check the big clusters BP analysis results with common pathways and unique pathways.

```{r, simGO function and basic files}
library(rrvgo)


simBP_fn <- function(GOdat, threshold=0.70, orgdb="org.Mm.eg.db", ont = "BP", fileName, width=6, height=5){
  simMatrix <- calculateSimMatrix(GOdat$ID,
                                orgdb=orgdb,
                                ont=ont,
                                method="Rel")

  scores <- setNames(-log10(GOdat$qvalue), GOdat$ID)
  reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=threshold,
                                orgdb=orgdb)
  pdf(fileName, width = width, height = height)
  scaplt <- scatterPlot(simMatrix, reducedTerms) 
  print(scaplt)
  dev.off()
  

}
macNames <- c("Resall", "IFNall", "AIRall", "Reswk3", "Reswk11", "Reswk20", "IFNwk3", "IFNwk11", "IFNwk20",
              "AIRwk3", "AIRwk11", "AIRwk20")

fileNames <- paste0("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/",macNames,"_GOSim_ScatterPlot_July_2023.pdf")
fileInput <- c(subAllList,GORes_BPT, GOIFN_BPT, GOAIR_BPT)
for(i in 1:length(fileInput)){
  splt <- simBP_fn(fileInput[[i]], threshold=0.90, orgdb="org.Mm.eg.db", ont = "BP", fileNames[i], width=6, height=5)
  splt
}
```
Check the common ones, First check the Adv/Res Macrophage across time.

```{r}
TcommonID_Res <- rownames(TimeResoverTab)[rowSums(TimeResoverTab)==3]
TcommonID_Res_wk3  <- GORes_BPT[[1]][GORes_BPT[[1]]$ID%in%TcommonID_Res,]
TcommonID_Res_wk11 <- GORes_BPT[[2]][GORes_BPT[[2]]$ID%in%TcommonID_Res,]
TcommonID_Res_wk20 <- GORes_BPT[[3]][GORes_BPT[[3]]$ID%in%TcommonID_Res,]
TcommonID_Res_list <- list(TcommonID_Res_wk3, TcommonID_Res_wk11, TcommonID_Res_wk20)
TcommonID_Res_Name <- paste0("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/SimGO_Res_common_wk", c(3, 11, 20), "_scatterPlot_July_2023.pdf")
for(i in 1:3){
  TcommonID_Res_plt <- simBP_fn(TcommonID_Res_list[[i]], threshold=0.90, orgdb="org.Mm.eg.db", ont = "BP", 
                                TcommonID_Res_Name[i], width=10, height=12)
}

```
Check the unique ones for Adv/Res across time.
```{r}
TuniID_Res_wk3   <- rownames(TimeResoverTab)[TimeResoverTab[,1]==1&TimeResoverTab[,2]==0&TimeResoverTab[,3]==0]
TuniID_Res_wk11  <- rownames(TimeResoverTab)[TimeResoverTab[,1]==0&TimeResoverTab[,2]==1&TimeResoverTab[,3]==0]
TuniID_Res_wk20  <- rownames(TimeResoverTab)[TimeResoverTab[,1]==0&TimeResoverTab[,2]==0&TimeResoverTab[,3]==1]

TuniID_Res_wk3  <- GORes_BPT[[1]][GORes_BPT[[1]]$ID%in%TuniID_Res_wk3,]
TuniID_Res_wk11 <- GORes_BPT[[2]][GORes_BPT[[2]]$ID%in%TuniID_Res_wk11,]
TuniID_Res_wk20 <- GORes_BPT[[3]][GORes_BPT[[3]]$ID%in%TuniID_Res_wk20,]
TuniID_Res_list <- list(TuniID_Res_wk3, TuniID_Res_wk11, TuniID_Res_wk20)
TuniID_Res_Name <- paste0("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/SimGO_Res_unique_wk", c(3, 11, 20), "_scatterPlot_July_2023.pdf")
for(i in 1:3){
  TcommonID_Res_plt <- simBP_fn(TuniID_Res_list[[i]], threshold=0.90, orgdb="org.Mm.eg.db", ont = "BP", 
                                TuniID_Res_Name[i], width=10, height=12)
}
## write output for all GO into one file.
write.xlsx(as.data.frame(GORes_BP), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = F, sheetName = "Res_all")
write.xlsx(as.data.frame(GOIFN_BP), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11_wk20_July_2023.xlsx", append = T, sheetName = "IFN_all")
write.xlsx(as.data.frame(GOAIR_BP), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "AIR_all")

write.xlsx(as.data.frame(GORes_BPT[[1]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "Res_wk3")
write.xlsx(as.data.frame(GORes_BPT[[2]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "Res_wk11")
write.xlsx(as.data.frame(GORes_BPT[[3]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "Res_wk20")

write.xlsx(as.data.frame(GOIFN_BPT[[1]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "IFN_wk3")
write.xlsx(as.data.frame(GOIFN_BPT[[2]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "IFN_wk11")
write.xlsx(as.data.frame(GOIFN_BPT[[3]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "IFN_wk20")
write.xlsx(as.data.frame(GOAIR_BPT[[1]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "AIR_wk3")
write.xlsx(as.data.frame(GOAIR_BPT[[2]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023.xlsx", append = T, sheetName = "AIR_wk11")
write.xlsx(as.data.frame(GOAIR_BPT[[3]]), file =
             "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_Res_IFN_AIR_wk3_wk11
           _wk20_July_2023_1.xlsx", append = T, sheetName = "AIR_wk20")


```

Perform common GO and unique GO across different subtypes macs and same type of mac across time.

```{r}
Common_Uni_fnT <- function(OverTM, TGOList, datNames, fileName){
  TcommonID <- rownames(OverTM)[rowSums(OverTM)==3]
  TcommonID_list <- lapply(TGOList, function(x) x[x$ID%in%TcommonID,])
  
  TuniID_1   <- rownames(OverTM)[OverTM[,1]==1&OverTM[,2]==0&OverTM[,3]==0]
  TuniID_2   <- rownames(OverTM)[OverTM[,1]==0&OverTM[,2]==1&OverTM[,3]==0]
  TuniID_3   <- rownames(OverTM)[OverTM[,1]==0&OverTM[,2]==0&OverTM[,3]==1]
  
  TuniID_dat1 <- TGOList[[1]][TGOList[[1]]$ID%in%TuniID_1,]
  TuniID_dat2 <- TGOList[[2]][TGOList[[2]]$ID%in%TuniID_2,]
  TuniID_dat3 <- TGOList[[3]][TGOList[[3]]$ID%in%TuniID_3,]
  
  Tcommon_merL <- lapply(TcommonID_list, function(x) x[,c("Description", "geneID")])
  Tcommon_mer  <- Reduce(function(dx,dy) merge(dx, dy, by = "Description"), Tcommon_merL)
  colnames(Tcommon_mer) <- c("Description", datNames)
  gc()
  write.xlsx(Tcommon_mer, file = fileName, append = F, sheetName="Common")
  write.xlsx(TuniID_dat1, file = fileName, append = T, sheetName=paste0("Uni_",datNames[1]))
  write.xlsx(TuniID_dat2, file = fileName, append = T, sheetName=paste0("Uni_",datNames[2]))
  write.xlsx(TuniID_dat3, file = fileName, append = T, sheetName=paste0("Uni_",datNames[3]))
  gc()
}
TfileNames <-  paste0("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_",
                      c("Res", "IFN", "AIR"), "_common_unique_wk3_wk11_wk20_July_2023.xlsx")
Common_Uni_fnT(OverTM=TimeResoverTab, TGOList=GORes_BPT, datNames=c("wk3", "wk11", "wk20"), TfileNames[1])
Common_Uni_fnT(OverTM=TimeIFNoverTab, TGOList=GOIFN_BPT, datNames=c("wk3", "wk11", "wk20"), TfileNames[2])
Common_Uni_fnT(OverTM=TimeAIRoverTab, TGOList=GOAIR_BPT, datNames=c("wk3", "wk11", "wk20"), TfileNames[3])

## different type of Mac all.
AfileNames <-  paste0("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/GeneOntology_BP_all_common_unique_all_July_2023.xlsx")
Common_Uni_fnT(OverTM=AlloverTab, TGOList=subAllList, datNames=c("Res", "IFN", "AIR"), AfileNames)
```
In order to make a straightforward comparison on biological process and Kegg pathway either across the time or across three subtype macrophages, apply the "compareCluster" function in "clusterProfiler" package.

```{r}
compareGO_fn <- function(datinput, Namesdat){
  dList <- datinput
  dList <- lapply(dList, function(x) x$ENTREZID)
  dList <- lapply(dList, function(x) as.character(x[!is.na(x)]))
  names(dList) <- Namesdat
  ck_BP <- compareCluster(geneCluster = dList, fun = "enrichGO", OrgDb = "org.Mm.eg.db")
  ck_BP <- setReadable(ck_BP, OrgDb = "org.Mm.eg.db", keyType="ENTREZID")
  ck_KEGG <- compareCluster(geneCluster = dList, fun = "enrichKEGG", organism="mmu")
  ck_KEGG <- setReadable(ck_KEGG, OrgDb = "org.Mm.eg.db", keyType="ENTREZID")
  ss <- list(ck_BP, ck_KEGG)
  ss
}
allGOinputDL_all3 <- list(allGOinputDL[[1]],allGOinputDL[[2]],allGOinputDL[[3]])
allck <- compareGO_fn(allGOinputDL_all3, c("Adv/Res", "IFN/Inflam", "AIR/Trem2"))
## compare the different macrophage across different time. 
Tall_data_Res <- list(allGOinputDL[[4]],  allGOinputDL[[5]], allGOinputDL[[6]])
Tall_data_IFN <- list(allGOinputDL[[7]],  allGOinputDL[[8]], allGOinputDL[[9]])
Tall_data_AIR <- list(allGOinputDL[[10]],  allGOinputDL[[11]], allGOinputDL[[12]])
allResT <- compareGO_fn(Tall_data_Res, c("Res_w3", "Res_w11", "Res_w20"))
allIFNT <- compareGO_fn(Tall_data_IFN, c("IFN_w3", "IFN_w11", "IFN_w20"))
allAIRT <- compareGO_fn(Tall_data_AIR, c("AIR_w3", "AIR_w11", "AIR_w20"))
## compare one time across three subtype macrophages.
T1data <- list(allGOinputDL[[4]],  allGOinputDL[[7]], allGOinputDL[[10]])
T2data <- list(allGOinputDL[[5]],  allGOinputDL[[8]], allGOinputDL[[11]])
T3data <- list(allGOinputDL[[6]],  allGOinputDL[[9]], allGOinputDL[[12]])
allT1  <- compareGO_fn(T1data, c("Res_w3", "IFN_w3", "AIR_w3"))
allT2  <- compareGO_fn(T2data, c("Res_w11", "IFN_w11", "AIR_w11"))
allT3  <- compareGO_fn(T3data, c("Res_w20", "IFN_w20", "AIR_w20"))

## all in one compare.
mydfList <- list(allGOinputDL[[4]],  allGOinputDL[[5]], allGOinputDL[[6]],
                 allGOinputDL[[7]],  allGOinputDL[[8]], allGOinputDL[[9]],
                 allGOinputDL[[10]],  allGOinputDL[[11]], allGOinputDL[[12]])
condList <- rep(c("Res", "IFN", "AIR"), each=3)
timeList <- rep(c("w3", "w11", "w20"), length = 9)
for(i in 1:length(mydfList)){
  mydfList[[i]]$condition <- condList[i]
  mydfList[[i]]$time      <- timeList[i]
  mydfList[[i]]           <- mydfList[[i]][!is.na(mydfList[[i]]$ENTREZID),]
 
}
alldfInput <- Reduce(rbind, mydfList)
formula_KEGGAll <- compareCluster(ENTREZID~condition+time, data=alldfInput, fun="enrichKEGG", organism = "mmu")
formula_BPAll   <- compareCluster(ENTREZID~condition+time, data=alldfInput, fun="enrichGO", OrgDb = "org.Mm.eg.db")
formula_KEGGAll <- setReadable(formula_KEGGAll, OrgDb = "org.Mm.eg.db", keyType="ENTREZID")
formula_BPAll   <- setReadable(formula_BPAll, OrgDb = "org.Mm.eg.db", keyType="ENTREZID")

save(formula_KEGGAll, formula_BPAll, allT1, allT2, allT3, allResT, allIFNT, allAIRT, allck, 
     file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/CompareCluster_BP_KEGG_allobject_July_2023.RData")

## double check the results and get some idea. 
  

```
#### The following is for checking,

Next, generate volcanoPlot and heatmap for selected markers under different comparisons.

```{r, fig.width = 14, fig.height = 4}

library(ggrepel)
functionPlotDEVolcano <- function(resfile, sig_cut, logfc_cut, title,  xrange, yrange, selgenes, xlabel, ylabel) {
  
  results       <- resfile
  results$genes <- rownames(results)
  results       <- subset(results, !is.na(p_val_adj))
  results       <- results[order(-results$avg_log2FC),]
  pmin          <- min(Res_MAST$p_val_adj[Res_MAST$p_val_adj!=0])
  results$p_val_adj <- ifelse(results$p_val_adj==0, pmin-0.0001, results$p_val_adj)
  datap         <- subset(results, avg_log2FC > logfc_cut & p_val_adj<= sig_cut)
  datan         <- subset(results, avg_log2FC < (-logfc_cut) & p_val_adj<= sig_cut)
  datap1        <- subset(results, avg_log2FC > 0 )
  datan1        <- subset(results, avg_log2FC < 0 )
  
  volc.plt <- ggplot(data=results, aes(x=avg_log2FC, y=-log10(p_val_adj), label=genes)) +
    geom_vline(xintercept = logfc_cut,     colour="black", linetype = "dashed", alpha=0.5) +
    geom_vline(xintercept = -(logfc_cut),  colour="black", linetype = "dashed", alpha=0.5) +
    geom_hline(yintercept = -log10(sig_cut), colour="black", linetype = "dashed", alpha=0.5) +
    geom_point(data=subset(results, abs(avg_log2FC) < logfc_cut | p_val_adj > sig_cut), alpha=0.75, size=0.3, colour="grey") +
    geom_point(data=subset(results, p_val_adj<=sig_cut & avg_log2FC >= logfc_cut),      alpha=0.75, size=0.8, colour="pink") +
    geom_point(data=subset(results, p_val_adj<=sig_cut & avg_log2FC <= -(logfc_cut)),   alpha=0.75, size=0.8, colour="cyan") +
    geom_point(data= datap1[datap1$gene%in%selgenes,], alpha = 0.75, size=1.8, colour = "red", shape = 24, fill = "red") +
    geom_point(data= datan1[datan1$gene%in%selgenes,], alpha = 0.75, size=1.8, colour = "blue", shape= 25, fill = "blue") +
    
    geom_text_repel( data= datap1[datap1$gene%in%selgenes,],fontface="bold.italic",
                     show.legend = FALSE, nudge_x=0.2, nudge_y=0.3, segment.size = 0.25, size=4) +
    geom_text_repel( data= datan1[datan1$gene%in%selgenes,],fontface="bold.italic",
                     show.legend = FALSE, nudge_x=0.2, nudge_y=0.3, segment.size = 0.25, size=4 ) +
    xlab(xlabel) + ylab(ylabel) + 
    #scale_x_continuous(limits=c(xrange[1],xrange[2]), breaks=seq(xrange[1],xrange[2],xrange[3])) +
    #scale_y_continuous(limits=c(yrange[1],yrange[2]), breaks=seq(yrange[1],yrange[2],yrange[3])) +
    theme(aspect.ratio=1) +
    ggtitle(title) +
    theme_update(plot.title = element_text(size=16, face="bold", hjust=0.5),
                 axis.title.x = element_text(size=12, face= "bold"),
                 axis.text.x = element_text(size=12, face="bold"),
                 axis.title.y.left = element_text(size=12, face= "bold"),
                 axis.text.y = element_text(size=12, face="bold")) +
    theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
          panel.background = element_rect(fill = "white", colour = NA)) 
  
  
  return(volc.plt)
  
}


sig_cut <- 0.05
logfc_cut <- 0.27
title <- "Adv/Res Mac"
xrange <- c(-2, 2, 0.5)
yrange <- c(0, 5, 1)
topN <- 20
xlabel    <- "log2FC (HFD/Ctrl)"
ylabel    <- "-log"[10] ~ "(adj.p.value)"

advallM <- functionPlotDEVolcano(Res_MAST, sig_cut, logfc_cut, 'Adv/Res ',  xrange, yrange, Res_markers, xlabel, ylabel)
IFNallM <- functionPlotDEVolcano(IFNI_MAST, sig_cut, logfc_cut, "IFN/Inflam",  xrange, yrange, IFN_markers, xlabel, ylabel)
AirallM <- functionPlotDEVolcano(AIRT_MAST, sig_cut, logfc_cut, "MacAir/Trem2",  xrange, yrange, AIR_markers, xlabel, ylabel)


## Time course with three groups
advt1M <- functionPlotDEVolcano(AdvR_Time_MAST[[1]], sig_cut, logfc_cut, "Res/Adv wk3",  xrange, yrange, Res_markers, xlabel, ylabel)
IFNt1M <- functionPlotDEVolcano(IFNI_Time_MAST[[1]], sig_cut, logfc_cut, "IFN/Inflam wk3",  xrange, yrange, IFN_markers, xlabel, ylabel)
Airt1M <- functionPlotDEVolcano(AIRT_Time_MAST[[1]], sig_cut, logfc_cut, "MacAir/Trem2 wk3",  xrange, yrange, AIR_markers, xlabel, ylabel)

advt2M <- functionPlotDEVolcano(AdvR_Time_MAST[[2]], sig_cut, logfc_cut, "Res/Adv wk11",  xrange, yrange, Res_markers, xlabel, ylabel)
IFNt2M <- functionPlotDEVolcano(IFNI_Time_MAST[[2]], sig_cut, logfc_cut, "IFN/Inflam wk11",  xrange, yrange, IFN_markers, xlabel, ylabel)
Airt2M <- functionPlotDEVolcano(AIRT_Time_MAST[[2]], sig_cut, logfc_cut, "MacAir/Trem2 wk11",  xrange, yrange, AIR_markers, xlabel, ylabel)

advt3M <- functionPlotDEVolcano(AdvR_Time_MAST[[3]], sig_cut, logfc_cut, "Res/Adv wk20",  xrange, yrange, Res_markers, xlabel, ylabel)
IFNt3M <- functionPlotDEVolcano(IFNI_Time_MAST[[3]], sig_cut, logfc_cut, "IFN/Inflam wk20",  xrange, yrange, IFN_markers, xlabel, ylabel)
Airt3M <- functionPlotDEVolcano(AIRT_Time_MAST[[3]], sig_cut, logfc_cut, "MacAir/Trem2 wk20",  xrange, yrange, AIR_markers, xlabel, ylabel)

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/TimeCourse_volplt_Macs_Res12_C19_merge_C3_July_2023.pdf", height = 8, width = 22)
plot_grid(advt1M, advt2M, advt3M, ncol = 3)
plot_grid(IFNt1M, IFNt2M, IFNt3M, ncol = 3)
plot_grid(Airt1M, Airt2M, Airt3M, ncol = 3)

dev.off()
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/ConditionAll_volplt_Macs_Res12_C19_merge_C3_July_2023.pdf", height = 6, width = 18)
plot_grid(advallM, IFNallM, AirallM, ncol = 3)
dev.off()

```
Try to just use macrophage to test HFDvsCtrl to show that what type of Macrophage enriched in HFD.
```{r}
Idents(testall) <- testall$NewCondition
testall@meta.data$Mac_cond <- ifelse(Idents(testall)=="Adv/Res_Chow"| Idents(testall)==
                                       "IFN_Chow"|Idents(testall)== "Inflam_Chow" 
                                     |Idents(testall)== "Mac_AIR/Trem2_Chow", "Mac_Chow",
                                     "Others")
testall@meta.data$Mac_cond <- ifelse(Idents(testall)=="Adv/Res_HFD"| Idents(testall)== "IFN_HFD"
                                     | Idents(testall)== "Inflam_HFD" | Idents(testall)==
                                       "Mac_AIR/Trem2_HFD","Mac_HFD", testall@meta.data$Mac_cond)
Idents(testall) <- testall@meta.data$Mac_cond 
detest_Mac <- FindMarkers(testall, assay = "RNA", ident.1 = "Mac_HFD", ident.2 = "Mac_Chow", verbose = FALSE, logfc.threshold = 0.00)
pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/MACall_HFDvsChow_volplt_Macs_June_2023.pdf", height = 6, width = 8)
mac_vplt <- functionPlotDEVolcano(detest_Mac, sig_cut, logfc_cut, "Macrophage",  xrange, yrange, selgenes, xlabel, ylabel)
mac_vplt
dev.off()
```
Need to discuss with Ziad to see what we would like to put into the main/supplementary, merged time or split time (heatmap/GO analysis.). Finished analysis at this stage on 10th July, 2023. And also how to present the GWAS causal genes GO analysis in the paper.

## New sections for Mouse macrophage analysis after 07112023

1) stacked barplot to show the time point HFD and Chow macrophage celltype distributed. 
2) How to present the GO analysis for HFDvChow and also across the time. 

```{r}
load("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/NewMAC_Reclustered_withConditions_meta_Robj_Res12_C19_GC3_GFC4_June_2023.RData")

barplt_datall <- as.data.frame(table(MAC_Reclustered@meta.data$Global_CCA_Annotation, MAC_Reclustered@meta.data$Condition))
barplt_datall_sub <- barplt_datall[-c(2,5:7, 9, 12:14), ]
barplt_datall_sub$Perc <- c(barplt_datall_sub$Freq[1:3]/sum(barplt_datall_sub$Freq[1:3])*100, 
                     barplt_datall_sub$Freq[4:6]/sum(barplt_datall_sub$Freq[4:6])*100)
colnames(barplt_datall_sub) <- c("CellType", "Treatment", "Freq", "Proportion")

barplt_datwk <- as.data.frame(table(MAC_Reclustered@meta.data$Global_CCA_Annotation, MAC_Reclustered@meta.data$TimeCond))
barplt_datwk_sub <- barplt_datwk[-c(1:7, 9, 12:14, 16, 19:21, 23, 26:28), ]
barplt_datwk_sub <- barplt_datwk_sub[c(7:9, 1:3, 4:6),]
barplt_datwk_sub$Perc <- c(barplt_datwk_sub$Freq[1:3]/sum(barplt_datwk_sub$Freq[1:3])*100, 
                     barplt_datwk_sub$Freq[4:6]/sum(barplt_datwk_sub$Freq[4:6])*100,
                     barplt_datwk_sub$Freq[7:9]/sum(barplt_datwk_sub$Freq[7:9])*100)
colnames(barplt_datwk_sub) <- c("CellType", "Treatment", "Freq", "Proportion")

barplt_dat <- rbind(barplt_datall_sub, barplt_datwk_sub)
barplt_dat$Treatment <- factor(barplt_dat$Treatment, levels = c("Chow", "HFD", "wk3_HFD", "wk11_HFD", "wk20_HFD"))
labels <- c("Chow", "HFD", "3wks\nHFD", "11wks\nHFD", "20wks\nHFD")
 write.csv(barplt_dat, file = "/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Figxx-Mouse_Mac_Proportion_HFDvsCtrl_Time_stackedBar_Data_Dec_2023.csv")

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Figxx-Mouse_Mac_Proportion_HFDvsCtrl_Time_StackedBar_07112023_version1.pdf", width = 8, height = 4)
barplt <- ggplot(barplt_dat, aes(fill=CellType, y=Proportion, x=Treatment)) + 
    geom_bar(position="fill", stat="identity") + 
    scale_x_discrete(label = labels) +
    scale_fill_manual(values=c("darkred",  "darkgray",  "darkorchid")) +
  theme(panel.background = element_blank()) +
  theme(axis.line = element_line(colour = "black"))
print(barplt)
dev.off()

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Figxx-Mouse_Mac_Proportion_HFDvsCtrl_Time_StackedBar_07112023_version1_update_29112023.pdf", width = 8, height = 4)
newbarplt_dat <- barplt_dat[-c(1:6),];
newlabels <- labels[c(5,4,3)]
newbarplt_dat$CellType <- factor(newbarplt_dat$CellType, levels = c( "IFN/Inflam", "Mac_AIR/Trem2", "Adv/Res"))
newbarplt_dat$Treatment <- factor(newbarplt_dat$Treatment, levels = c("wk20_HFD", "wk11_HFD", "wk3_HFD"))
newbarplt <- ggplot(newbarplt_dat, aes(fill=CellType, y=Proportion, x=Treatment)) + 
    geom_bar(position="fill", stat="identity") + 
    scale_x_discrete(label = newlabels) +
    scale_fill_manual(values=c( "darkgray", "darkmagenta",  "darkred")) +
  theme(panel.background = element_blank()) +
  theme(axis.line = element_line(colour = "black")) + coord_flip()
print(newbarplt)
dev.off()
## version 2 is to joint two barplot, Chow and HFD, and then based on HFD proportion plot each cell type for time.
barplt_datwk_sub_v2 <- barplt_datwk_sub[c(3,6,9,2,5,8,1,4,7),]
barplt_datwk_sub_v2$CellType <- factor(barplt_datwk_sub_v2$CellType, levels = c( "Mac_AIR/Trem2", "IFN/Inflam", "Adv/Res"))
barplt_datwk_sub_v2$Treatment <- factor(barplt_datwk_sub_v2$Treatment, levels = c( "wk20_HFD", "wk11_HFD", "wk3_HFD"))

barplt_datwk_sub_v2$Proportion <- c(barplt_datwk_sub_v2[c(1:3),3]/sum(barplt_datwk_sub_v2[c(1:3),3]),
                                    barplt_datwk_sub_v2[c(4:6),3]/sum(barplt_datwk_sub_v2[c(4:6),3]),
                                    barplt_datwk_sub_v2[c(7:9),3]/sum(barplt_datwk_sub_v2[c(7:9),3]))



barplt_s1 <- ggplot(barplt_dat[1:6,], aes(fill=CellType, y=Proportion, x=Treatment)) + 
    geom_bar(position="fill", stat="identity") + 
    scale_x_discrete(label = labels) +
    scale_fill_manual(values=c("darkred",  "darkgray",  "darkorchid")) +
  theme(panel.background = element_blank()) +
  theme(axis.line = element_line(colour = "black"),legend.position="bottom")
barplt_s2 <- ggplot(barplt_datwk_sub_v2, aes(fill=Treatment, y=Proportion, x=CellType)) + 
    geom_bar(position="fill", stat="identity") + 
    #scale_x_discrete(label = labels) +
    scale_fill_manual(values=c( "chocolate", "gold",  "cyan" )) +
  theme(panel.background = element_blank()) +
  theme(axis.line = element_line(colour = "black"), legend.position="bottom") +coord_flip()

pdf("/Users/xz289/Documents/Minoru/Seurat_v3/Athero_mouse/Athero_mouse_LDLR_new/Figxx-Mouse_Mac_Proportion_HFDvsCtrl_Time_StackedBar_07112023_version2.pdf", width = 13, height = 4)
plot_grid(barplt_s1, barplt_s2, ncol=2, rel_widths = c(0.9, 1.5))
dev.off()

```


